<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jessica Cooperstone">

<title>Principal Components Analysis â€“ ðŸ’»ðŸ§°ðŸ“ŠðŸ¥³</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../img/code-slash.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-647f65b01dad988ac54bf3b04a0860e4.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-06d2be3c2e19c0a3f6003c9d0585ee8c.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-M24LDNCMG6"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-M24LDNCMG6', { 'anonymize_ip': true});
</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../favicon.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">ðŸ’»ðŸ§°ðŸ“ŠðŸ¥³</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-syllabus" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Syllabus</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-syllabus">    
        <li>
    <a class="dropdown-item" href="../../../syllabus/syllabus.html">
 <span class="dropdown-text">Syllabus</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-course-content" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Course content</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-course-content">    
        <li class="dropdown-header">Module 1 - Principles</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module1/01_principles/01_principles.html">
 <span class="dropdown-text">Week 1 - Principles</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module1/02_good-and-bad/02_goodbad.html">
 <span class="dropdown-text">Week 2 - Good and bad visualizations</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd.html">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling.html">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101.html">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102.html">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions.html">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations.html">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats.html">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca.html">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_interactive-plots/11_interactive-plots.html">
 <span class="dropdown-text">Week 11 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_manhattan/12_manhattan.html">
 <span class="dropdown-text">Week 12 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers.html">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-recitations" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Recitations</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-recitations">    
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd_recitation.html">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation.html">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation.html">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102_recitation.html">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions_recitation.html">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations_recitation.html">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats_recitation.html">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca_recitation.html">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_interactive-plots/11_interactive-plots_recitation.html">
 <span class="dropdown-text">Week 11 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_manhattan/12_manhattan_recitation.html">
 <span class="dropdown-text">Week 12 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers_recitation.html">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-assignments" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Assignments</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-assignments">    
        <li class="dropdown-header">Module assignments</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module1/module_1_assignment.html">
 <span class="dropdown-text">Module 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module2/module_2_assignment.html">
 <span class="dropdown-text">Module 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module3/module_3_assignment.html">
 <span class="dropdown-text">Module 3</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module4/module_4_assignment.html">
 <span class="dropdown-text">Module 4</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Reflections</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/reflections.html">
 <span class="dropdown-text">Reflection instructions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Recitations</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/recitations.html">
 <span class="dropdown-text">Recitation instructions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Capstone assignment</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/capstone.html">
 <span class="dropdown-text">Capstone instructions</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-solutions" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Solutions</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-solutions">    
        <li class="dropdown-header">Module assignment solutions</li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Recitation solutions</li>
        <li>
    <a class="dropdown-item" href="https://raw.githubusercontent.com/jcooperstone/jcooperstone.github.io/main/rmds/03_Rmd_recitation_solutions.Rmd">
 <span class="dropdown-text">Week 3 - R Markdown solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation_solutions.html">
 <span class="dropdown-text">Week 4 - Wrangling solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation_solutions.html">
 <span class="dropdown-text">Week 5 - ggplot101 solutions</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/jcooperstone/rdataviz" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../modules/module4/10_pca/10_pca.html">Module 4 - Putting it together</a></li><li class="breadcrumb-item"><a href="../../../modules/module4/10_pca/10_pca.html">Week 10 - Principal components analysis</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 1 - Principles</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/01_principles/01_principles.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 1 - Principles</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/02_good-and-bad/02_goodbad.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 2 - Good and bad visualizations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 2 - Coding fundamentals</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/03_rmarkdown/03_Rmd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 3 - R Markdown</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/04_wrangling/04_wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 4 - Wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/05_ggplot101/05_ggplot101.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 5 - ggplot101</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/06_ggplot102/06_ggplot102.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 6 - ggplot102</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 3 - Data exploration</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/07_distributions/07_distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 7 - Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/08_correlations/08_correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 8 - Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/09_add-stats/09_add-stats.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 9 - Adding statistics</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 4 - Putting it together</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/10_pca/10_pca.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Week 10 - Principal components analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/11_interactive-plots/11_interactive-plots.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 11 - Interactive plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/12_manhattan/12_manhattan.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 12 - Manhattan plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/13_leftovers/13_leftovers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 13 - Leftovers</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#light-wrangling" id="toc-light-wrangling" class="nav-link" data-scroll-target="#light-wrangling">Light wrangling</a></li>
  </ul></li>
  <li><a href="#run-pca" id="toc-run-pca" class="nav-link" data-scroll-target="#run-pca">Run PCA</a></li>
  <li><a href="#scree-plot" id="toc-scree-plot" class="nav-link" data-scroll-target="#scree-plot">Scree plot</a>
  <ul>
  <li><a href="#using-fviz_eig" id="toc-using-fviz_eig" class="nav-link" data-scroll-target="#using-fviz_eig">Using <code>fviz_eig()</code></a></li>
  <li><a href="#manually" id="toc-manually" class="nav-link" data-scroll-target="#manually">Manually</a></li>
  </ul></li>
  <li><a href="#scores-plot" id="toc-scores-plot" class="nav-link" data-scroll-target="#scores-plot">Scores plot</a>
  <ul>
  <li><a href="#using-fviz_pca_ind" id="toc-using-fviz_pca_ind" class="nav-link" data-scroll-target="#using-fviz_pca_ind">Using <code>fviz_pca_ind()</code></a></li>
  <li><a href="#manually-1" id="toc-manually-1" class="nav-link" data-scroll-target="#manually-1">Manually</a></li>
  </ul></li>
  <li><a href="#loadings-plot" id="toc-loadings-plot" class="nav-link" data-scroll-target="#loadings-plot">Loadings plot</a>
  <ul>
  <li><a href="#using-fviz_pca_var" id="toc-using-fviz_pca_var" class="nav-link" data-scroll-target="#using-fviz_pca_var">Using <code>fviz_pca_var()</code></a></li>
  <li><a href="#manually-2" id="toc-manually-2" class="nav-link" data-scroll-target="#manually-2">Manually</a></li>
  </ul></li>
  <li><a href="#patchwork" id="toc-patchwork" class="nav-link" data-scroll-target="#patchwork"><code>patchwork</code></a></li>
  <li><a href="#biplot" id="toc-biplot" class="nav-link" data-scroll-target="#biplot">Biplot</a>
  <ul>
  <li><a href="#using-fviz_pca." id="toc-using-fviz_pca." class="nav-link" data-scroll-target="#using-fviz_pca.">Using <code>fviz_pca()</code>.</a></li>
  <li><a href="#manually-3" id="toc-manually-3" class="nav-link" data-scroll-target="#manually-3">Manually</a></li>
  </ul></li>
  <li><a href="#useful-resources" id="toc-useful-resources" class="nav-link" data-scroll-target="#useful-resources">Useful resources</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../modules/module4/10_pca/10_pca.html">Module 4 - Putting it together</a></li><li class="breadcrumb-item"><a href="../../../modules/module4/10_pca/10_pca.html">Week 10 - Principal components analysis</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Principal Components Analysis</h1>
<p class="subtitle lead">Week 10</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jessica Cooperstone </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>Today we are going to start Module 4 where we put together a lot of the material weâ€™ve learned in the first 3 modules of this course. Todayâ€™s material is on conducting principal components analysis (PCA) using R, and visualizing the results with some tools weâ€™ve already learned to use, and some new wrangling and viz tips along the way.</p>
<p>PCA is a data reduction approach, and useful if you have many variables, for example, thousands of genes or metabolites. PCA creates summary variables (the principal components) which maximize the variation in the dataset. It can be categorized as an unsupervised approach, as PCA doesnâ€™t know which samples belong to your different groups. When you look at a scores plot, points that are closer together are more similar based on your input data, and those further apart are more different. The location of the loadings helps you understand what is driving those differences in your scores plot.</p>
<p>If you are unfamiliar with PCA, Iâ€™d recommend these two <a href="https://www.youtube.com/c/joshstarmer">Youtube videos by Josh Starmer of StatQuest</a> which <a href="https://www.youtube.com/watch?v=HMOI_lkzW08">explain PCA in 5 mins</a>, or <a href="https://www.youtube.com/watch?v=FgakZw6K1QQ">with more detail in 20 min</a>. Bam ðŸ’¥!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># everything</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl) <span class="co"># reading in excel sheets</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># easy PCA plotting</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue) <span class="co"># easy pasting</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="read-in-data" class="level4">
<h4 class="anchored" data-anchor-id="read-in-data">Read in data</h4>
<p>Today we are going to continue to use the same real research data from my group from the lessons on distributions and correlations. We will be reading in the supplementary data from a <a href="https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/tpg2.20192">paper</a> from my group written by Michael Dzakovich, and published in The Plant Genome. The data is present in a Excel worksheet, so we will use the function <code>read_excel()</code> from the tidyverse (but not core tidyverse) package <code>readxl</code>. We want to import Supplemental Table 3. You can indicate which sheet you want to import in the arguments to <code>read_excel()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>alkaloid_blups <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"data/tpg220192-sup-0002-supmat.xlsx"</span>,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">sheet =</span> <span class="st">"S3 BLUP Diversity Panel"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Letâ€™s take a look at this new data sheet.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(alkaloid_blups))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Genotype</th>
<th style="text-align: left;">Plot_Source</th>
<th style="text-align: left;">Species</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Origin</th>
<th style="text-align: left;">Provence</th>
<th style="text-align: left;">Blanca_Cluster1</th>
<th style="text-align: left;">Blanca_Cluster2</th>
<th style="text-align: left;">Passport_Species</th>
<th style="text-align: left;">Passport_Classification</th>
<th style="text-align: left;">Sim_Grouping</th>
<th style="text-align: right;">Dehydrotomatidine</th>
<th style="text-align: right;">Tomatidine</th>
<th style="text-align: right;">DehydrotomatineA1</th>
<th style="text-align: right;">Dehydrotomatine2</th>
<th style="text-align: right;">TotalDehydrotomatine</th>
<th style="text-align: right;">Tomatine</th>
<th style="text-align: right;">Hydroxytomatine1</th>
<th style="text-align: right;">Hydroxytomatine2</th>
<th style="text-align: right;">Hydroxytomatine3</th>
<th style="text-align: right;">Hydroxytomatine4</th>
<th style="text-align: right;">TotalHydroxytomatine</th>
<th style="text-align: right;">Acetoxytomatine1</th>
<th style="text-align: right;">Acetoxytomatine2</th>
<th style="text-align: right;">Acetoxytomatine3</th>
<th style="text-align: right;">TotalAcetoxytomatine</th>
<th style="text-align: right;">DehydrolycoperosideFGdehydroesculeosideA</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA1</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA2</th>
<th style="text-align: right;">TotalLycoperosideFGEsculeosideA</th>
<th style="text-align: right;">EsculeosideB1</th>
<th style="text-align: right;">EsculeosideB2</th>
<th style="text-align: right;">EsculeosideB3</th>
<th style="text-align: right;">TotalEsculeosideB</th>
<th style="text-align: right;">Total</th>
<th style="text-align: left;">Latitude</th>
<th style="text-align: left;">Longitude</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K9-8584</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: right;">0.0001504</td>
<td style="text-align: right;">0.0043495</td>
<td style="text-align: right;">0.0072241</td>
<td style="text-align: right;">-0.0010738</td>
<td style="text-align: right;">0.0061503</td>
<td style="text-align: right;">0.1929106</td>
<td style="text-align: right;">0.0102521</td>
<td style="text-align: right;">0.1596566</td>
<td style="text-align: right;">0.0352666</td>
<td style="text-align: right;">0.0864042</td>
<td style="text-align: right;">0.2917962</td>
<td style="text-align: right;">-0.0680508</td>
<td style="text-align: right;">-0.1242448</td>
<td style="text-align: right;">0.0117481</td>
<td style="text-align: right;">-0.1805475</td>
<td style="text-align: right;">-0.0369856</td>
<td style="text-align: right;">0.0056573</td>
<td style="text-align: right;">-0.1693034</td>
<td style="text-align: right;">-0.1636462</td>
<td style="text-align: right;">-0.0340685</td>
<td style="text-align: right;">-0.0061713</td>
<td style="text-align: right;">-0.0371543</td>
<td style="text-align: right;">-0.0773942</td>
<td style="text-align: right;">0.0365648</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
</tr>
<tr class="even">
<td style="text-align: left;">CULBPT_05_15</td>
<td style="text-align: left;">2K9-8622</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: right;">-0.0000272</td>
<td style="text-align: right;">-0.0001716</td>
<td style="text-align: right;">0.0086547</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0086547</td>
<td style="text-align: right;">0.2824926</td>
<td style="text-align: right;">0.0006910</td>
<td style="text-align: right;">0.1118818</td>
<td style="text-align: right;">0.0243950</td>
<td style="text-align: right;">0.0189345</td>
<td style="text-align: right;">0.1559021</td>
<td style="text-align: right;">0.0038232</td>
<td style="text-align: right;">0.0467380</td>
<td style="text-align: right;">0.0024737</td>
<td style="text-align: right;">0.0530349</td>
<td style="text-align: right;">0.0066251</td>
<td style="text-align: right;">0.0125334</td>
<td style="text-align: right;">0.3417664</td>
<td style="text-align: right;">0.3542998</td>
<td style="text-align: right;">0.0138429</td>
<td style="text-align: right;">0.0021833</td>
<td style="text-align: right;">0.0139309</td>
<td style="text-align: right;">0.0299571</td>
<td style="text-align: right;">0.8907677</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CULBPT_05_22</td>
<td style="text-align: left;">2K17-7708-1</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: right;">-0.0000649</td>
<td style="text-align: right;">-0.0002516</td>
<td style="text-align: right;">0.0085205</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0085205</td>
<td style="text-align: right;">0.1755250</td>
<td style="text-align: right;">-0.0001804</td>
<td style="text-align: right;">0.0505547</td>
<td style="text-align: right;">0.0040900</td>
<td style="text-align: right;">0.0023229</td>
<td style="text-align: right;">0.0576202</td>
<td style="text-align: right;">0.0078240</td>
<td style="text-align: right;">0.0069610</td>
<td style="text-align: right;">-0.0001145</td>
<td style="text-align: right;">0.0146705</td>
<td style="text-align: right;">0.0031468</td>
<td style="text-align: right;">0.0150674</td>
<td style="text-align: right;">0.3840689</td>
<td style="text-align: right;">0.3991363</td>
<td style="text-align: right;">0.0007128</td>
<td style="text-align: right;">0.0008227</td>
<td style="text-align: right;">0.0028133</td>
<td style="text-align: right;">0.0043489</td>
<td style="text-align: right;">0.6618186</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
</tr>
<tr class="even">
<td style="text-align: left;">CULBPT04_1</td>
<td style="text-align: left;">2K9-8566</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: right;">-0.0000272</td>
<td style="text-align: right;">0.0001259</td>
<td style="text-align: right;">-0.0038737</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">-0.0038737</td>
<td style="text-align: right;">-0.0061446</td>
<td style="text-align: right;">0.0015930</td>
<td style="text-align: right;">0.1059454</td>
<td style="text-align: right;">0.0195829</td>
<td style="text-align: right;">0.0083566</td>
<td style="text-align: right;">0.1354779</td>
<td style="text-align: right;">0.0098585</td>
<td style="text-align: right;">0.0095344</td>
<td style="text-align: right;">0.0007941</td>
<td style="text-align: right;">0.0201870</td>
<td style="text-align: right;">0.0049377</td>
<td style="text-align: right;">0.0100416</td>
<td style="text-align: right;">0.3466282</td>
<td style="text-align: right;">0.3566699</td>
<td style="text-align: right;">0.0100830</td>
<td style="text-align: right;">0.0006432</td>
<td style="text-align: right;">0.0089015</td>
<td style="text-align: right;">0.0196277</td>
<td style="text-align: right;">0.5269806</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
</tr>
<tr class="odd">
<td style="text-align: left;">E6203</td>
<td style="text-align: left;">2K9-8600</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">CA</td>
<td style="text-align: left;">SLL_processing_1</td>
<td style="text-align: left;">SLL_processing_1_1</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_CA</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: right;">-0.0000272</td>
<td style="text-align: right;">0.0000159</td>
<td style="text-align: right;">0.0099538</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0099538</td>
<td style="text-align: right;">0.2929606</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0227075</td>
<td style="text-align: right;">0.0039160</td>
<td style="text-align: right;">0.0088279</td>
<td style="text-align: right;">0.0354514</td>
<td style="text-align: right;">0.0002365</td>
<td style="text-align: right;">0.0395374</td>
<td style="text-align: right;">0.0041794</td>
<td style="text-align: right;">0.0439532</td>
<td style="text-align: right;">0.0027439</td>
<td style="text-align: right;">0.0133225</td>
<td style="text-align: right;">0.3018732</td>
<td style="text-align: right;">0.3151958</td>
<td style="text-align: right;">0.0080894</td>
<td style="text-align: right;">-0.0003574</td>
<td style="text-align: right;">0.0100045</td>
<td style="text-align: right;">0.0177365</td>
<td style="text-align: right;">0.7179839</td>
<td style="text-align: left;">36.778300000000002</td>
<td style="text-align: left;">-119.4179</td>
</tr>
<tr class="even">
<td style="text-align: left;">F06-2041</td>
<td style="text-align: left;">2K16-9843</td>
<td style="text-align: left;">Processing</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">OH</td>
<td style="text-align: left;">SLL_processing_1</td>
<td style="text-align: left;">SLL_processing_1_3</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_OH</td>
<td style="text-align: left;">Humid</td>
<td style="text-align: right;">-0.0000272</td>
<td style="text-align: right;">-0.0001648</td>
<td style="text-align: right;">-0.0011236</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">-0.0011236</td>
<td style="text-align: right;">0.0169681</td>
<td style="text-align: right;">0.0011237</td>
<td style="text-align: right;">0.0195573</td>
<td style="text-align: right;">-0.0016308</td>
<td style="text-align: right;">-0.0016069</td>
<td style="text-align: right;">0.0188454</td>
<td style="text-align: right;">-0.0004237</td>
<td style="text-align: right;">0.0271524</td>
<td style="text-align: right;">-0.0005286</td>
<td style="text-align: right;">0.0262002</td>
<td style="text-align: right;">0.0053084</td>
<td style="text-align: right;">0.0049033</td>
<td style="text-align: right;">0.1647694</td>
<td style="text-align: right;">0.1696728</td>
<td style="text-align: right;">-0.0006298</td>
<td style="text-align: right;">-0.0003074</td>
<td style="text-align: right;">0.0019315</td>
<td style="text-align: right;">0.0009943</td>
<td style="text-align: right;">0.2352713</td>
<td style="text-align: left;">40.417299999999997</td>
<td style="text-align: left;">-82.9071</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>What are the dimensions of this dataframe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(alkaloid_blups)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 107  37</code></pre>
</div>
</div>
</section>
<section id="light-wrangling" class="level3">
<h3 class="anchored" data-anchor-id="light-wrangling">Light wrangling</h3>
<p>Here we have the best linear unbiased predictors (BLUPs) representing the alkaloid content of 107 genotypes of tomatoes. There is extra meta-data here we wonâ€™t use, so like we did in correlations, we are going to create a vector to indicate which column name reprents the alkaloids we want to include in our principal components analysis. Then we can create a new trimmed dataframe.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>alkaloid_total_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Dehydrotomatidine"</span>,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Tomatidine"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalDehydrotomatine"</span>,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Tomatine"</span>,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalHydroxytomatine"</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalAcetoxytomatine"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"DehydrolycoperosideFGdehydroesculeosideA"</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalLycoperosideFGEsculeosideA"</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalEsculeosideB"</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Total"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>alkaloid_blups_trim <span class="ot">&lt;-</span> alkaloid_blups <span class="sc">|&gt;</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Genotype, Species, Class, <span class="fu">all_of</span>(alkaloid_total_names))</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># did it work?</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(alkaloid_blups_trim) <span class="co"># yes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Genotype"                                
 [2] "Species"                                 
 [3] "Class"                                   
 [4] "Dehydrotomatidine"                       
 [5] "Tomatidine"                              
 [6] "TotalDehydrotomatine"                    
 [7] "Tomatine"                                
 [8] "TotalHydroxytomatine"                    
 [9] "TotalAcetoxytomatine"                    
[10] "DehydrolycoperosideFGdehydroesculeosideA"
[11] "TotalLycoperosideFGEsculeosideA"         
[12] "TotalEsculeosideB"                       
[13] "Total"                                   </code></pre>
</div>
</div>
</section>
</section>
<section id="run-pca" class="level2">
<h2 class="anchored" data-anchor-id="run-pca">Run PCA</h2>
<p>There are many packages that have functions that run PCA (including ) but I think the most common function used is a part of base R, and is called <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/prcomp"><code>prcomp()</code></a>.</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note, PCA will allow zeroes, but will throw an error if you feed it NAs.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>alkaloids_pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(alkaloid_blups_trim <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="fu">all_of</span>(alkaloid_total_names)),</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">scale =</span> <span class="cn">TRUE</span>, <span class="co"># default is false</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">center =</span> <span class="cn">TRUE</span>) <span class="co"># default is true, just being explicit</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Letâ€™s investigate <code>alkaloids_pca</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 5
 $ sdev    : num [1:10] 1.794 1.732 1.215 0.99 0.776 ...
 $ rotation: num [1:10, 1:10] 0.162 0.309 0.422 0.326 0.311 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:10] "Dehydrotomatidine" "Tomatidine" "TotalDehydrotomatine" "Tomatine" ...
  .. ..$ : chr [1:10] "PC1" "PC2" "PC3" "PC4" ...
 $ center  : Named num [1:10] 0.000169 0.00721 0.142798 1.865975 1.323755 ...
  ..- attr(*, "names")= chr [1:10] "Dehydrotomatidine" "Tomatidine" "TotalDehydrotomatine" "Tomatine" ...
 $ scale   : Named num [1:10] 0.000505 0.024096 0.339379 5.889986 2.91824 ...
  ..- attr(*, "names")= chr [1:10] "Dehydrotomatidine" "Tomatidine" "TotalDehydrotomatine" "Tomatine" ...
 $ x       : num [1:107, 1:10] -1.51 -1.5 -1.55 -1.55 -1.52 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : NULL
  .. ..$ : chr [1:10] "PC1" "PC2" "PC3" "PC4" ...
 - attr(*, "class")= chr "prcomp"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Standard deviations (1, .., p=10):
 [1] 1.7944141686 1.7318715307 1.2151400892 0.9904494488 0.7763914595
 [6] 0.6695313752 0.4394142511 0.2041300753 0.1932182462 0.0001503975

Rotation (n x k) = (10 x 10):
                                               PC1         PC2         PC3
Dehydrotomatidine                        0.1621938  0.05363402 -0.06746652
Tomatidine                               0.3088504 -0.13094437 -0.44243776
TotalDehydrotomatine                     0.4222731 -0.30216776 -0.22116400
Tomatine                                 0.3263804 -0.28919346 -0.43424991
TotalHydroxytomatine                     0.3111090 -0.07515005  0.42780515
TotalAcetoxytomatine                     0.3190534 -0.17396967  0.54287286
DehydrolycoperosideFGdehydroesculeosideA 0.2125680  0.49447329 -0.07346836
TotalLycoperosideFGEsculeosideA          0.2130280  0.52056383 -0.08463631
TotalEsculeosideB                        0.1864604  0.50165801 -0.10122926
Total                                    0.5191805  0.04443498  0.24834278
                                                 PC4          PC5         PC6
Dehydrotomatidine                         0.92897283 -0.275272461 -0.09845544
Tomatidine                                0.13150651  0.482582095  0.65879800
TotalDehydrotomatine                     -0.16555038 -0.170452131 -0.15777601
Tomatine                                 -0.13638091 -0.163532836 -0.41927007
TotalHydroxytomatine                      0.15203796  0.706002906 -0.41591986
TotalAcetoxytomatine                     -0.05462895 -0.326082234  0.41704857
DehydrolycoperosideFGdehydroesculeosideA -0.16278360  0.013643330 -0.06919125
TotalLycoperosideFGEsculeosideA          -0.09078436 -0.009476836 -0.02310538
TotalEsculeosideB                         0.02495982 -0.037371129  0.01529154
Total                                    -0.11066018 -0.170588385  0.05600981
                                                 PC7         PC8          PC9
Dehydrotomatidine                         0.13097525 -0.02201625 -0.012909093
Tomatidine                                0.03505366  0.05778857  0.054315300
TotalDehydrotomatine                      0.03480557 -0.61023340 -0.475709677
Tomatine                                 -0.09303946  0.40353582  0.403874861
TotalHydroxytomatine                     -0.04702814 -0.04023463  0.009165434
TotalAcetoxytomatine                     -0.01377758 -0.02258828  0.182475550
DehydrolycoperosideFGdehydroesculeosideA  0.61727735 -0.31823996  0.437114636
TotalLycoperosideFGEsculeosideA           0.10012665  0.42335087 -0.583833541
TotalEsculeosideB                        -0.76029219 -0.28461575  0.203734859
Total                                    -0.01570365  0.31194871 -0.025467058
                                                  PC10
Dehydrotomatidine                        -0.0001604947
Tomatidine                                0.0010655927
TotalDehydrotomatine                      0.0147128823
Tomatine                                  0.2559833151
TotalHydroxytomatine                      0.1268345182
TotalAcetoxytomatine                      0.5059522400
DehydrolycoperosideFGdehydroesculeosideA  0.0080094872
TotalLycoperosideFGEsculeosideA           0.3707973486
TotalEsculeosideB                         0.0226436777
Total                                    -0.7239562737</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "prcomp"</code></pre>
</div>
</div>
<p>We can see that the resulting PCA object is a prcomp object, and is a list of 5 lists and vectors.</p>
<p>This includes:</p>
<ul>
<li><code>sdev</code>: the standard deviations (square roots of the eigenvalues of the covariance matrix) of the principal components</li>
<li><code>rotation</code>: the PCs for the variables (i.e., the variable loadings)</li>
<li><code>x</code>: the PCs for samples (i.e., the scores)</li>
<li><code>center</code>: the centering used</li>
<li><code>scale</code>: the scaling used</li>
</ul>
<p>We can also look at the output of our PCA in a different way using the function <code>summary()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(alkaloids_pca) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Importance of components:
                         PC1    PC2    PC3    PC4     PC5     PC6     PC7
Standard deviation     1.794 1.7319 1.2151 0.9904 0.77639 0.66953 0.43941
Proportion of Variance 0.322 0.2999 0.1477 0.0981 0.06028 0.04483 0.01931
Cumulative Proportion  0.322 0.6219 0.7696 0.8677 0.92796 0.97279 0.99210
                           PC8     PC9      PC10
Standard deviation     0.20413 0.19322 0.0001504
Proportion of Variance 0.00417 0.00373 0.0000000
Cumulative Proportion  0.99627 1.00000 1.0000000</code></pre>
</div>
</div>
<p>We can convert this summary into something later usable by extraction the element <code>importance</code> from <code>summary(alkaloids_pca)</code> and converting it to a dataframe.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>importance <span class="ot">&lt;-</span> <span class="fu">summary</span>(alkaloids_pca)<span class="sc">$</span>importance <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(importance))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 6%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">PC1</th>
<th style="text-align: right;">PC2</th>
<th style="text-align: right;">PC3</th>
<th style="text-align: right;">PC4</th>
<th style="text-align: right;">PC5</th>
<th style="text-align: right;">PC6</th>
<th style="text-align: right;">PC7</th>
<th style="text-align: right;">PC8</th>
<th style="text-align: right;">PC9</th>
<th style="text-align: right;">PC10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard deviation</td>
<td style="text-align: right;">1.794414</td>
<td style="text-align: right;">1.731872</td>
<td style="text-align: right;">1.21514</td>
<td style="text-align: right;">0.9904494</td>
<td style="text-align: right;">0.7763915</td>
<td style="text-align: right;">0.6695314</td>
<td style="text-align: right;">0.4394143</td>
<td style="text-align: right;">0.2041301</td>
<td style="text-align: right;">0.1932182</td>
<td style="text-align: right;">0.0001504</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proportion of Variance</td>
<td style="text-align: right;">0.321990</td>
<td style="text-align: right;">0.299940</td>
<td style="text-align: right;">0.14766</td>
<td style="text-align: right;">0.0981000</td>
<td style="text-align: right;">0.0602800</td>
<td style="text-align: right;">0.0448300</td>
<td style="text-align: right;">0.0193100</td>
<td style="text-align: right;">0.0041700</td>
<td style="text-align: right;">0.0037300</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Cumulative Proportion</td>
<td style="text-align: right;">0.321990</td>
<td style="text-align: right;">0.621930</td>
<td style="text-align: right;">0.76959</td>
<td style="text-align: right;">0.8676900</td>
<td style="text-align: right;">0.9279600</td>
<td style="text-align: right;">0.9727900</td>
<td style="text-align: right;">0.9921000</td>
<td style="text-align: right;">0.9962700</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>By looking at the summary we can see, for example, that the first two PCs explain 62.19% of variance.</p>
<p>We are going to go over making scree, scores and loadings plots using helper functions (here, they start <code>fviz_()</code> and come from the package <a href="https://www.rdocumentation.org/packages/factoextra/versions/1.0.7"><code>factoextra</code></a>, and manually via ggplot. The helper functions allow you look at each plot type simply. This is an important step because when you make your plots with ggplot, you want to be sure they look how they should.</p>
</section>
<section id="scree-plot" class="level2">
<h2 class="anchored" data-anchor-id="scree-plot">Scree plot</h2>
<p>A <a href="https://en.wikipedia.org/wiki/Scree_plot#:~:text=In%20multivariate%20statistics%2C%20a%20scree,principal%20component%20analysis%20(PCA).">scree plot</a> shows what percentage of total variance is explained by each principal component.</p>
<section id="using-fviz_eig" class="level3">
<h3 class="anchored" data-anchor-id="using-fviz_eig">Using <code>fviz_eig()</code></h3>
<p>We can do this quickly using <a href="https://rpkgs.datanovia.com/factoextra/reference/eigenvalue.html"><code>fviz_eig()</code></a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_eig</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/fviz_eig-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>You can actually do this very easily with base R plotting as well. If you werenâ€™t planning to publish this type of plot, it might not be important it look beautiful, and then both of these options would be great and quick. Note though that the base R plot is plotting at a different scale.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/plot alkaloids_pca-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="manually" class="level3">
<h3 class="anchored" data-anchor-id="manually">Manually</h3>
<p>If you wanted to make a scree plot manually, you could by plotting using a wrangled version of the <code>importance</code> dataframe we made earlier.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>importance_tidy <span class="ot">&lt;-</span> importance <span class="sc">|&gt;</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"measure"</span>) <span class="sc">|&gt;</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> PC1<span class="sc">:</span>PC10,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"PC"</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"value"</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>importance_tidy <span class="sc">|&gt;</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(measure <span class="sc">==</span> <span class="st">"Proportion of Variance"</span>) <span class="sc">|&gt;</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y  =</span> value)) <span class="sc">+</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/importance wrangling manual plot 1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Almost! PC10 is displaying right after PC1 because alphabetically, this is the order. Letâ€™s fix it.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a vector with the order we want</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>my_order <span class="ot">&lt;-</span> <span class="fu">colnames</span>(importance)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># relevel according to my_order</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>importance_tidy<span class="sc">$</span>PC <span class="ot">&lt;-</span> <span class="fu">factor</span>(importance_tidy<span class="sc">$</span>PC, <span class="at">levels =</span> my_order)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># check to see if it worked</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(importance_tidy<span class="sc">$</span>PC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "PC1"  "PC2"  "PC3"  "PC4"  "PC5"  "PC6"  "PC7"  "PC8"  "PC9"  "PC10"</code></pre>
</div>
</div>
<p>Letâ€™s plot again.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>importance_tidy <span class="sc">|&gt;</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(measure <span class="sc">==</span> <span class="st">"Proportion of Variance"</span>) <span class="sc">|&gt;</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y  =</span> value)) <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/scree plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Success!</p>
<p>If we want to tighten up this plot we can.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>importance_tidy <span class="sc">|&gt;</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(measure <span class="sc">==</span> <span class="st">"Proportion of Variance"</span>) <span class="sc">|&gt;</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y  =</span> value)) <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Principal component"</span>,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Percent variance explained"</span>,</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Scree plot of 10 alkaloids analyzed across 107 tomato accessions"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/scree plot prettier-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is a perfectly ready scree plot for the supplementary materials of a publication.</p>
</section>
</section>
<section id="scores-plot" class="level2">
<h2 class="anchored" data-anchor-id="scores-plot">Scores plot</h2>
<p>When people talk about PCA plots, what they most often mean is PCA scores plots. Here, each point represents a sample, and we are plotting their coordinates typically for the first 2 PCs. Sometimes people make 3D PCA plots with the first 3 PCs but I think these are not easy to look in 2D and I wouldnâ€™t recommend you to put them in your papers.</p>
<section id="using-fviz_pca_ind" class="level4">
<h4 class="anchored" data-anchor-id="using-fviz_pca_ind">Using <code>fviz_pca_ind()</code></h4>
<p>We can also look at a scores plot using <a href="https://rpkgs.datanovia.com/factoextra/reference/fviz_pca.html"><code>fviz_pca_ind()</code></a> where ind means individuals. Here, each point is a sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_ind</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/scores fviz_pca_ind-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Because our <code>alkaloids_pca</code> doesnâ€™t have any meta-data, this is a hard to interpret plot, where each number indicates the rownumber of that sample. Making the scores plot this way is useful because it shows us the shape of the plot which we can use to confirm that we have made a ggplot that looks like its been created correctly.</p>
</section>
<section id="manually-1" class="level4">
<h4 class="anchored" data-anchor-id="manually-1">Manually</h4>
<p>We want to plot the scores, which are in provided in <code>alkaloids_pca$x</code>.</p>
<p>We can convert the list into a dataframe of scores values by using <code>as.data.frame()</code>. Then we can bind back our relevant metadata so theyâ€™re all together. Note, to use <code>bind_cols()</code> both datasets need to be in the same order. In this case they are so we are good.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a df of alkaloids_pca$x</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>scores_raw <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(alkaloids_pca<span class="sc">$</span>x)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># bind meta-data</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(alkaloid_blups <span class="sc">|&gt;</span> <span class="fu">select</span>(Genotype, Plot_Source, Species), <span class="co"># metadata</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>                    scores_raw)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># how does our new df look?</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>scores[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 Ã— 6
  Genotype     Plot_Source Species      PC1    PC2     PC3
  &lt;chr&gt;        &lt;chr&gt;       &lt;chr&gt;      &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
1 CULBPT_05_11 2K9-8584    Processing -1.51 -1.17  0.00632
2 CULBPT_05_15 2K9-8622    Processing -1.50 -0.915 0.0649 
3 CULBPT_05_22 2K17-7708-1 Processing -1.55 -0.942 0.0658 
4 CULBPT04_1   2K9-8566    Processing -1.55 -0.906 0.0817 
5 E6203        2K9-8600    Processing -1.52 -0.940 0.0436 
6 F06-2041     2K16-9843   Processing -1.58 -0.934 0.0677 </code></pre>
</div>
</div>
<p>Now we can plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">|&gt;</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/scores plotting 1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Our shapes are looking the same, this is good. Letâ€™s pretty up our plot.</p>
<p>First letâ€™s wrangle.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create objects indicating percent variance explained by PC1 and PC2</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>PC1_percent <span class="ot">&lt;-</span> <span class="fu">round</span>((importance[<span class="dv">2</span>,<span class="dv">1</span>])<span class="sc">*</span><span class="dv">100</span>, <span class="co"># index 2nd row, 1st column, times 100</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                     <span class="dv">1</span>) <span class="co"># round to 1 decimal</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>PC2_percent <span class="ot">&lt;-</span> <span class="fu">round</span>((importance[<span class="dv">2</span>,<span class="dv">2</span>])<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>) </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co"># make Species a factor and set levels</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co"># from least wild to most wild</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co"># we did this back in data distributions.</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>scores<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="fu">factor</span>(scores<span class="sc">$</span>Species,</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Processing"</span>,</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"Cultivated Cherry"</span>,</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"Hybrid"</span>,</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"Wild Cherry"</span>,</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"Wild"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then we can plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>(scores_plot <span class="ot">&lt;-</span> scores <span class="sc">|&gt;</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">fill =</span> Species)) <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>   <span class="co"># sequential color selecting hex codes from YlOrRd color by wildness</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#ffffcc"</span>, <span class="st">"#fed976"</span>, <span class="st">"#fd8d3c"</span>, <span class="st">"#e31a1c"</span>, <span class="st">"#800026"</span>)) <span class="sc">+</span> </span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Scores Plot of 10 Alkaloids Present in 107 Tomato Accessions"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This looks nice.</p>
</section>
</section>
<section id="loadings-plot" class="level2">
<h2 class="anchored" data-anchor-id="loadings-plot">Loadings plot</h2>
<section id="using-fviz_pca_var" class="level3">
<h3 class="anchored" data-anchor-id="using-fviz_pca_var">Using <code>fviz_pca_var()</code></h3>
<p>We can also look at a loadings plot using <a href="https://rpkgs.datanovia.com/factoextra/reference/fviz_pca.html"><code>fviz_pca_var()</code></a> where var means variables. Here, each point is a variable.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_var</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/loadings fviz_pca_var-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="manually-2" class="level3">
<h3 class="anchored" data-anchor-id="manually-2">Manually</h3>
<p>We can also make a more customized loadings plot manually using ggplot and using the dataframe <code>alkaloids_pca$rotation</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># grab raw loadings, without any metadata</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>loadings_raw <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(alkaloids_pca<span class="sc">$</span>rotation)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>loadings <span class="ot">&lt;-</span> loadings_raw <span class="sc">|&gt;</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"alkaloid"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can then plot with ggplot like normal.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>loadings <span class="sc">|&gt;</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> alkaloid)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>() <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Loadings Plot"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/loadings plot 1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We have two problems with this plot.</p>
<ol type="1">
<li>The names are abbreviated and not how we want them to appear</li>
<li>The label names are on top of the points/each other</li>
</ol>
<p>We can fix both of these problems.</p>
<p>We can create a vector of the labels as we want them to appear, as we have done previously.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>alkaloid_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Dehydrotomatidine"</span>,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Tomatidine"</span>,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Dehydrotomatine"</span>,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Alpha-Tomatine"</span>,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Hydroxytomatine"</span>,</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Acetoxytomatine"</span>,</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Dehydrlycoperoside F, G, </span><span class="sc">\n</span><span class="st">or Dehydroescueloside A"</span>,</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lycoperoside F, G, </span><span class="sc">\n</span><span class="st">or Escueloside A"</span>,</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Escueloside B"</span>,</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Total Steroidal Alkaloids"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then we can re-plot with these labels.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>loadings <span class="sc">|&gt;</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> alkaloid_labels)) <span class="sc">+</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span>  </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>() <span class="sc">+</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Loadings Plot"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/loadings plot 2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok the label names are better but theyâ€™re still smushed. The package <a href="https://ggrepel.slowkow.com/index.html"><code>ggrepel</code></a> has some good functions to help us. You can try using <a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html"><code>geom_text_repel()</code></a> and <a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html"><code>geom_label_repel()</code></a>.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="img/ggrepel_cartoon.jpeg" class="img-fluid figure-img" style="width:70.0%" alt="A cartoon that says ggrepel, wonderfully repulsive labels, and has some fuzzy monsters and a sewing theme making a plot with buttons, and labels that are repelled away indicating what color they are. There is a ball of yarn indicating segments, and patches indicating labels."></p>
<figcaption>Artwork by <a href="https://twitter.com/allison_horst">Allison Horst</a></figcaption>
</figure>
</div>
</div>
</div>
<p>With <code>geom_text_repel()</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>(loadings_plot <span class="ot">&lt;-</span> loadings <span class="sc">|&gt;</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> alkaloid_labels)) <span class="sc">+</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span>  </span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>() <span class="sc">+</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>() <span class="sc">+</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Loadings Plot using geom_text_repel()"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/geom_text_repel-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>With <code>geom_label_repel()</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>loadings <span class="sc">|&gt;</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> alkaloid_labels)) <span class="sc">+</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>() <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>() <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Loadings Plot using geom_label_repel()"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/geom_label_repel-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="patchwork" class="level2">
<h2 class="anchored" data-anchor-id="patchwork"><code>patchwork</code></h2>
<p>You can pop these two plots side by side easing using the package <a href="https://patchwork.data-imaginist.com/"><code>patchwork</code></a>.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="img/patchwork.jpeg" class="img-fluid figure-img" style="width:70.0%" alt="Fuzzy cartoon monsters in white gloves and uniforms hanging multiple plots together on a wall, with an artist monster wearing a beret and smock directing them to the correct orientation. There is a blueprint plan on the wall showing how the plots should be arranged. Stylized title font reads â€œpatchwork - combine &amp; arrange your ggplots!â€"></p>
<figcaption>Artwork by <a href="https://twitter.com/allison_horst">Allison Horst</a></figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>scores_plot <span class="sc">+</span> loadings_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/pathwork 1-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="biplot" class="level2">
<h2 class="anchored" data-anchor-id="biplot">Biplot</h2>
<section id="using-fviz_pca." class="level3">
<h3 class="anchored" data-anchor-id="using-fviz_pca.">Using <code>fviz_pca()</code>.</h3>
<p>You can make a biplot quickly with <a href="https://rpkgs.datanovia.com/factoextra/reference/fviz_pca.html"><code>fviz_pca()</code></a>. Note, <code>fviz_pca_biplot()</code> and <code>fviz_pca()</code> are the same.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca</span>(alkaloids_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/fviz_pca-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Instead of making this plot manually, letâ€™s go through how to alter the existing plot made with <code>fviz_pca()</code>. We can do this because <code>factoextra</code> creates ggplot objects. To start off, we need to be using a dataframe that includes our metadata.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca</span>(alkaloids_pca, <span class="co"># pca object</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">label =</span> <span class="st">"var"</span>,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">geom.var =</span> <span class="st">"text"</span>) <span class="sc">+</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> alkaloid_blups<span class="sc">$</span>Species), <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set2"</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Biplot Plot of 10 Alkaloids Present in 107 Tomato Accessions"</span>,</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/fviz_pca altered 1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is almost what we want - except we have only the abbreviated names for the alkaloids. Since in a biplot, we are really plotting two different sets of data (the scores and the loadings)there isnâ€™t the ability to use <code>labeller</code> or similar with <code>fviz_pca</code> for the loadings only. There is a workaround though, we can go into our PCA object, change the rownames of <code>alkaloids_pca$rotation</code> to be our longer labels, and that should inherit to our new plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save as a new df</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>alkaloids_pca_labelled <span class="ot">&lt;-</span> alkaloids_pca</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># assign alkaloid_labels to rownames</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(alkaloids_pca_labelled<span class="sc">$</span>rotation) <span class="ot">&lt;-</span> alkaloid_labels</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca</span>(alkaloids_pca_labelled, <span class="co"># pca object</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">label =</span> <span class="st">"var"</span>,</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">geom.var =</span> <span class="fu">c</span>(<span class="st">"text"</span>, <span class="st">"point"</span>),</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">col.var =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> alkaloid_blups<span class="sc">$</span>Species), <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set2"</span>) <span class="sc">+</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Biplot Plot of 10 Alkaloids Present in 107 Tomato Accessions"</span>,</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/fviz_pca relabelled plot 2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="manually-3" class="level3">
<h3 class="anchored" data-anchor-id="manually-3">Manually</h3>
<p>We canâ€™t add the loadings right on top of the scores because they are on different scales.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">|&gt;</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">fill =</span> Species), <span class="co"># from piped data</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> loadings,</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2)) <span class="sc">+</span> <span class="co"># set locally</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>   <span class="co"># sequential color selecting hex codes from YlOrRd color by wildness</span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#ffffcc"</span>, <span class="st">"#fed976"</span>, <span class="st">"#fd8d3c"</span>, <span class="st">"#e31a1c"</span>, <span class="st">"#800026"</span>)) <span class="sc">+</span> </span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Scores Plot of 10 Alkaloids Present in 107 Tomato Accessions"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The issue here is that the scores and loadings are on different scales - letâ€™s make them on the same scale.</p>
<p>I can write a quick function to allow normalization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>normalize <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">min</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then I can nornalize the scores using the scale function, since the loadings are already normalized.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>scores_normalized <span class="ot">&lt;-</span> scores <span class="sc">|&gt;</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PC1_norm =</span> <span class="fu">scale</span>(<span class="fu">normalize</span>(PC1), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>)) <span class="sc">|&gt;</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PC2_norm =</span> <span class="fu">scale</span>(<span class="fu">normalize</span>(PC2), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>)) <span class="sc">|&gt;</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Genotype, Plot_Source, Species, </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>         PC1_norm, PC2_norm, <span class="fu">everything</span>()) <span class="co"># pick metadata and reorder</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>How did it go? PC1_norm and PC2_norm should all now be between -1 and 1</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># look at first six rows</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(scores_normalized)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 Ã— 15
  Genotype    Plot_Source Species PC1_norm[,1] PC2_norm[,1]   PC1    PC2     PC3
  &lt;chr&gt;       &lt;chr&gt;       &lt;fct&gt;          &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
1 CULBPT_05_â€¦ 2K9-8584    Procesâ€¦       -0.201      -0.114  -1.51 -1.17  0.00632
2 CULBPT_05_â€¦ 2K9-8622    Procesâ€¦       -0.200      -0.0892 -1.50 -0.915 0.0649 
3 CULBPT_05_â€¦ 2K17-7708-1 Procesâ€¦       -0.206      -0.0918 -1.55 -0.942 0.0658 
4 CULBPT04_1  2K9-8566    Procesâ€¦       -0.206      -0.0883 -1.55 -0.906 0.0817 
5 E6203       2K9-8600    Procesâ€¦       -0.203      -0.0916 -1.52 -0.940 0.0436 
6 F06-2041    2K16-9843   Procesâ€¦       -0.210      -0.0910 -1.58 -0.934 0.0677 
# â„¹ 7 more variables: PC4 &lt;dbl&gt;, PC5 &lt;dbl&gt;, PC6 &lt;dbl&gt;, PC7 &lt;dbl&gt;, PC8 &lt;dbl&gt;,
#   PC9 &lt;dbl&gt;, PC10 &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># range of PC1</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(scores_normalized<span class="sc">$</span>PC1_norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.2128768  0.7871232</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># range of PC2</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(scores_normalized<span class="sc">$</span>PC2_norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.4608813  0.5391187</code></pre>
</div>
</div>
<p>We can set the <code>scores_normalized</code> data for the overall plot and add the <code>loadings</code> within another <code>geom_point()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>scores_normalized <span class="sc">|&gt;</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1_norm, <span class="at">y =</span> PC2_norm, <span class="at">fill =</span> Species), <span class="co"># scores</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span> </span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> loadings, <span class="co"># loadings points</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2), <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> loadings, <span class="co"># loadings labels</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> alkaloid_labels),</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">size =</span> <span class="fl">2.5</span>, <span class="co"># make labels smaller </span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fill =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">"white"</span>),<span class="fl">0.5</span>)) <span class="sc">+</span> <span class="co"># make label background transparent</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#ffffcc"</span>, <span class="st">"#fed976"</span>, <span class="st">"#fd8d3c"</span>, <span class="st">"#e31a1c"</span>, <span class="st">"#800026"</span>)) <span class="sc">+</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">glue</span>(<span class="st">"PC1: {PC1_percent}%"</span>), </span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">glue</span>(<span class="st">"PC2: {PC2_percent}%"</span>), </span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Biplot of 10 Alkaloids Present in 107 Tomato Accessions"</span>,</span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Samples are colored points and loadings are grey with labels"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10_pca_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Voila.</p>
</section>
</section>
<section id="useful-resources" class="level2">
<h2 class="anchored" data-anchor-id="useful-resources">Useful resources</h2>
<ul>
<li><a href="https://biodash.github.io/codeclub/s03e03_pca/">Code club about PCA by Jelmer Poelstra</a></li>
<li><a href="https://www.rdocumentation.org/packages/factoextra/versions/1.0.7">factoextra</a></li>
<li><a href="https://www.datacamp.com/tutorial/pca-analysis-r">PCA from DataCamp</a></li>
</ul>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/www\.rdataviz\.com\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Created with RStudio and <a href="https://quarto.org/">Quarto</a> <br> Â© Jessica Cooperstone, 2025, License: <a href="https://creativecommons.org/licenses/by-nc/4.0/">CC BY-NC 4.0</a> <br> <a href="https://osu-codeclub.github.io/#category=website">Want to make a website like this one?</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/jcooperstone/rdataviz">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>