<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jessica Cooperstone">

<title>💻🧰📊🥳 - Understanding Data Distributions</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../img/code-slash.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../favicon.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">💻🧰📊🥳</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-syllabus" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Syllabus</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-syllabus">    
        <li>
    <a class="dropdown-item" href="../../../syllabus/syllabus.html" rel="" target="">
 <span class="dropdown-text">Syllabus</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-course-content" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Course content</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-course-content">    
        <li class="dropdown-header">Module 1 - Principles</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module1/01_principles/01_principles.html" rel="" target="">
 <span class="dropdown-text">Week 1 - Principles</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module1/02_good-and-bad/02_goodbad.html" rel="" target="">
 <span class="dropdown-text">Week 2 - Good and bad visualizations</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd.html" rel="" target="">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling.html" rel="" target="">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101.html" rel="" target="">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102.html" rel="" target="">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions.html" rel="" target="">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations.html" rel="" target="">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats.html" rel="" target="">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca.html" rel="" target="">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_manhattan/11_manhattan.html" rel="" target="">
 <span class="dropdown-text">Week 11 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_interactive-plots/12_interactive-plots.html" rel="" target="">
 <span class="dropdown-text">Week 12 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers.html" rel="" target="">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-recitations" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Recitations</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-recitations">    
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_manhattan/11_manhattan_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 11 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_interactive-plots/12_interactive-plots_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 12 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers_recitation.html" rel="" target="">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-assignments" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Assignments</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-assignments">    
        <li class="dropdown-header">Module assignments</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module1/module_1_assignment.html" rel="" target="">
 <span class="dropdown-text">Module 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module2/module_2_assignment.html" rel="" target="">
 <span class="dropdown-text">Module 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module3/module_3_assignment.html" rel="" target="">
 <span class="dropdown-text">Module 3</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module4/module_4_assignment.html" rel="" target="">
 <span class="dropdown-text">Module 4</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Reflections</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/reflections.html" rel="" target="">
 <span class="dropdown-text">Reflection instructions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Capstone assignment</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/capstone.html" rel="" target="">
 <span class="dropdown-text">Capstone instructions</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-solutions" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Solutions</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-solutions">    
        <li class="dropdown-header">Module assignment solutions</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module2/module_2_solutions.html" rel="" target="">
 <span class="dropdown-text">Module 2 solutions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Recitation solutions</li>
        <li>
    <a class="dropdown-item" href="https://github.com/jcooperstone/rdataviz/raw/main/rmds/03_Rmd_recitation_solutions.Rmd" rel="" target="">
 <span class="dropdown-text">Week 3 - R Markdown solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation_solutions.html" rel="" target="">
 <span class="dropdown-text">Week 4 - Wrangling solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation_solutions.html" rel="" target="">
 <span class="dropdown-text">Week 5 - ggplot101 solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102_recitation_solutions.html" rel="" target="">
 <span class="dropdown-text">Week 6 - ggplot102 solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions_recitation_solutions.html" rel="" target="">
 <span class="dropdown-text">Week 7 - Data distributions solutions</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <div class="quarto-navbar-tools ms-auto">
    <a href="https://github.com/jcooperstone/rdataviz" rel="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../modules/module3/07_distributions/07_distributions.html">Module 3 - Data exploration</a></li><li class="breadcrumb-item"><a href="../../../modules/module3/07_distributions/07_distributions.html">Week 7 - Data distributions</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Module 1 - Principles</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/01_principles/01_principles.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 1 - Principles</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/02_good-and-bad/02_goodbad.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 2 - Good and bad visualizations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Module 2 - Coding fundamentals</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/03_rmarkdown/03_Rmd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 3 - R Markdown</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/04_wrangling/04_wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 4 - Wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/05_ggplot101/05_ggplot101.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 5 - ggplot101</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/06_ggplot102/06_ggplot102.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 6 - ggplot102</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Module 3 - Data exploration</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/07_distributions/07_distributions.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Week 7 - Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/08_correlations/08_correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 8 - Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/09_add-stats/09_add-stats.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 9 - Adding statistics</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Module 4 - Putting it together</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/10_pca/10_pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 10 - Principal components analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/11_manhattan/11_manhattan.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 11 - Manhattan plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/12_interactive-plots/12_interactive-plots.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 12 - Interactive plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/13_leftovers/13_leftovers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 13 - Leftovers</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul>
  <li><a href="#load-libraries-and-data" id="toc-load-libraries-and-data" class="nav-link" data-scroll-target="#load-libraries-and-data">Load libraries and data</a></li>
  </ul></li>
  <li><a href="#geoms-for-distributions" id="toc-geoms-for-distributions" class="nav-link" data-scroll-target="#geoms-for-distributions">Geoms for distributions</a>
  <ul>
  <li><a href="#geom_col" id="toc-geom_col" class="nav-link" data-scroll-target="#geom_col"><code>geom_col()</code></a></li>
  <li><a href="#stat_summary" id="toc-stat_summary" class="nav-link" data-scroll-target="#stat_summary"><code>stat_summary()</code></a>
  <ul class="collapse">
  <li><a href="#reordering-x-variables" id="toc-reordering-x-variables" class="nav-link" data-scroll-target="#reordering-x-variables">Reordering x-variables</a></li>
  </ul></li>
  <li><a href="#geom_boxplot" id="toc-geom_boxplot" class="nav-link" data-scroll-target="#geom_boxplot"><code>geom_boxplot()</code></a></li>
  <li><a href="#geom_jitter" id="toc-geom_jitter" class="nav-link" data-scroll-target="#geom_jitter"><code>geom_jitter()</code></a></li>
  <li><a href="#geom_histogram" id="toc-geom_histogram" class="nav-link" data-scroll-target="#geom_histogram"><code>geom_histogram()</code></a></li>
  <li><a href="#ggridgesgeom_density_ridges" id="toc-ggridgesgeom_density_ridges" class="nav-link" data-scroll-target="#ggridgesgeom_density_ridges"><code>gg:ridges::geom_density_ridges()</code></a>
  <ul class="collapse">
  <li><a href="#changing-class-labels" id="toc-changing-class-labels" class="nav-link" data-scroll-target="#changing-class-labels">Changing class labels</a></li>
  </ul></li>
  <li><a href="#ggdist-functions" id="toc-ggdist-functions" class="nav-link" data-scroll-target="#ggdist-functions"><code>ggdist</code> functions</a></li>
  </ul></li>
  <li><a href="#useful-resources" id="toc-useful-resources" class="nav-link" data-scroll-target="#useful-resources">Useful resources</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Understanding Data Distributions</h1>
<p class="subtitle lead">Week 7</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jessica Cooperstone </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="img/summary_statistics.png" class="img-fluid figure-img" style="width:70.0%" alt="A cartoon showing a clip art style bar graph, on the left where all the data point dots are held at the bottom by a net, and one on the right where the fidelity of the datapoints are shown. The center says 'are your summary statistics hiding something interesting?'"></p>
<figcaption class="figure-caption">Figure from <a href="https://github.com/allisonhorst/stats-illustrations/blob/main/other-stats-artwork/summary_statistics.png">Allison Horst</a></figcaption>
</figure>
</div>
</div>
</div>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>We will will building on our lesson on ggplot101 and ggplot102 which focused on an overall understanding of the grammar of graphics, basic syntax, adding data, aesthetic mappings, geoms, facets, scales, labels, and themes. Today we are going to apply what we learned towards trying to better understanding our underlying data distributions.</p>
<p>Often, we think about figure generation as the last part of the scientific process, something you do as you prepare a manuscript for publication. I hope to convince you that exploring your data, and making exploratory plots is a critical part of the data analysis and interpretation process.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="img/ggplot2_exploratory.png" class="img-fluid figure-img" style="width:70.0%" alt="A group of fuzzy round monsters with binoculars, backpacks and guide books looking up a graphs flying around with wings (like birders, but with exploratory data visualizations). Stylized text reads “ggplot2: visual data exploration.”"></p>
<figcaption class="figure-caption">Figure from <a href="https://github.com/allisonhorst/stats-illustrations">Allison Horst</a></figcaption>
</figure>
</div>
</div>
</div>
<section id="load-libraries-and-data" class="level3">
<h3 class="anchored" data-anchor-id="load-libraries-and-data">Load libraries and data</h3>
<p>Before we get started, let’s load our libraries.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Today we are using real research data from my group. We will be reading in the supplementary data from a <a href="https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/tpg2.20192">paper</a> written by Michael Dzakovich, and published in The Plant Genome. The data is present in a Excel worksheet, so we will use the function <code>read_excel()</code> from the tidyverse (but not core tidyverse) package <code>readxl</code>. We want to import Supplemental Table 1. You can indicate which sheet you want to import in the arguments to <code>read_excel()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"tpg220192-sup-0002-supmat.xlsx"</span>,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">sheet =</span> <span class="st">"S1 Raw Data Diversity Panel"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(alkaloids))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">ID</th>
<th style="text-align: right;">Year</th>
<th style="text-align: left;">Environment</th>
<th style="text-align: right;">Block</th>
<th style="text-align: left;">Genotype</th>
<th style="text-align: left;">Plot_Source</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Origin</th>
<th style="text-align: left;">Provence</th>
<th style="text-align: left;">Blanca_Cluster1</th>
<th style="text-align: left;">Blanca_Cluster2</th>
<th style="text-align: left;">Passport_Species</th>
<th style="text-align: left;">Passport_Classification</th>
<th style="text-align: left;">Sim_Grouping</th>
<th style="text-align: left;">Latitude</th>
<th style="text-align: left;">Longitude</th>
<th style="text-align: right;">Dehydrotomatidine</th>
<th style="text-align: right;">Tomatidine</th>
<th style="text-align: right;">Dehydrotomatine1</th>
<th style="text-align: right;">Dehydrotomatine2</th>
<th style="text-align: right;">TotalDehydrotomatine</th>
<th style="text-align: right;">Tomatine</th>
<th style="text-align: right;">Hydroxytomatine1</th>
<th style="text-align: right;">Hydroxytomatine2</th>
<th style="text-align: right;">Hydroxytomatine3</th>
<th style="text-align: right;">Hydroxytomatine4</th>
<th style="text-align: right;">TotalHydroxytomatine</th>
<th style="text-align: right;">Acetoxytomatine1</th>
<th style="text-align: right;">Acetoxytomatine2</th>
<th style="text-align: right;">Acetoxytomatine3</th>
<th style="text-align: right;">TotalAcetoxytomatine</th>
<th style="text-align: right;">DehydrolycoperosideFGdehydroesculeosideA</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA1</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA2</th>
<th style="text-align: right;">TotalLycoperosideFGEsculeosideA</th>
<th style="text-align: right;">EsculeosideB1</th>
<th style="text-align: right;">EsculeosideB2</th>
<th style="text-align: right;">EsculeosideB3</th>
<th style="text-align: right;">TotalEsculeosideB</th>
<th style="text-align: right;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">7805</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreEarly18</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">5.726010</td>
<td style="text-align: right;">0.350331</td>
<td style="text-align: right;">6.076341</td>
<td style="text-align: right;">172.66244</td>
<td style="text-align: right;">1.079190</td>
<td style="text-align: right;">86.72742</td>
<td style="text-align: right;">17.831892</td>
<td style="text-align: right;">9.142607</td>
<td style="text-align: right;">114.78111</td>
<td style="text-align: right;">18.902399</td>
<td style="text-align: right;">56.307182</td>
<td style="text-align: right;">1.890053</td>
<td style="text-align: right;">77.099634</td>
<td style="text-align: right;">5.125904</td>
<td style="text-align: right;">10.277325</td>
<td style="text-align: right;">336.8893</td>
<td style="text-align: right;">347.1666</td>
<td style="text-align: right;">3.787979</td>
<td style="text-align: right;">0.924195</td>
<td style="text-align: right;">3.943230</td>
<td style="text-align: right;">8.655404</td>
<td style="text-align: right;">731.5675</td>
</tr>
<tr class="even">
<td style="text-align: right;">7898</td>
<td style="text-align: right;">2017</td>
<td style="text-align: left;">Fre17</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K9-8584</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.169068</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">55.47329</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">53.32292</td>
<td style="text-align: right;">13.630697</td>
<td style="text-align: right;">4.841762</td>
<td style="text-align: right;">71.79538</td>
<td style="text-align: right;">3.557348</td>
<td style="text-align: right;">4.107289</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">7.664637</td>
<td style="text-align: right;">2.905500</td>
<td style="text-align: right;">5.548102</td>
<td style="text-align: right;">199.6694</td>
<td style="text-align: right;">205.2175</td>
<td style="text-align: right;">8.978931</td>
<td style="text-align: right;">1.897850</td>
<td style="text-align: right;">6.794690</td>
<td style="text-align: right;">17.671471</td>
<td style="text-align: right;">360.8969</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7523</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreLate18</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.135675</td>
<td style="text-align: right;">0.680554</td>
<td style="text-align: right;">5.073552</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">5.073552</td>
<td style="text-align: right;">123.85835</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">50.90989</td>
<td style="text-align: right;">6.503939</td>
<td style="text-align: right;">1.368847</td>
<td style="text-align: right;">58.78268</td>
<td style="text-align: right;">3.931461</td>
<td style="text-align: right;">4.123222</td>
<td style="text-align: right;">0.623340</td>
<td style="text-align: right;">8.678023</td>
<td style="text-align: right;">2.185082</td>
<td style="text-align: right;">5.104115</td>
<td style="text-align: right;">259.0177</td>
<td style="text-align: right;">264.1218</td>
<td style="text-align: right;">4.049145</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">6.749386</td>
<td style="text-align: right;">10.798531</td>
<td style="text-align: right;">474.3143</td>
</tr>
<tr class="even">
<td style="text-align: right;">7724</td>
<td style="text-align: right;">2017</td>
<td style="text-align: left;">Fre17</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K9-8584</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.054300</td>
<td style="text-align: right;">0.497261</td>
<td style="text-align: right;">19.419087</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">19.419087</td>
<td style="text-align: right;">239.01264</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">36.02318</td>
<td style="text-align: right;">8.557673</td>
<td style="text-align: right;">7.483933</td>
<td style="text-align: right;">52.06478</td>
<td style="text-align: right;">3.341048</td>
<td style="text-align: right;">16.415426</td>
<td style="text-align: right;">1.057100</td>
<td style="text-align: right;">20.813574</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">203.0061</td>
<td style="text-align: right;">203.0061</td>
<td style="text-align: right;">1.678210</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">2.349633</td>
<td style="text-align: right;">4.027843</td>
<td style="text-align: right;">538.8955</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7427</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreLate18</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.139454</td>
<td style="text-align: right;">0.553801</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">64.31783</td>
<td style="text-align: right;">0.879435</td>
<td style="text-align: right;">39.91027</td>
<td style="text-align: right;">7.228388</td>
<td style="text-align: right;">3.015298</td>
<td style="text-align: right;">51.03339</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.131685</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.131685</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">4.054211</td>
<td style="text-align: right;">299.5687</td>
<td style="text-align: right;">303.6229</td>
<td style="text-align: right;">10.146857</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">4.882339</td>
<td style="text-align: right;">15.029197</td>
<td style="text-align: right;">437.8283</td>
</tr>
<tr class="even">
<td style="text-align: right;">7854</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreEarly18</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.049700</td>
<td style="text-align: right;">0.262174</td>
<td style="text-align: right;">3.737579</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.737579</td>
<td style="text-align: right;">68.44913</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">23.86864</td>
<td style="text-align: right;">13.506299</td>
<td style="text-align: right;">1.456982</td>
<td style="text-align: right;">38.83192</td>
<td style="text-align: right;">4.657902</td>
<td style="text-align: right;">4.259007</td>
<td style="text-align: right;">0.605729</td>
<td style="text-align: right;">9.522638</td>
<td style="text-align: right;">9.832149</td>
<td style="text-align: right;">11.595595</td>
<td style="text-align: right;">459.5205</td>
<td style="text-align: right;">471.1161</td>
<td style="text-align: right;">6.839930</td>
<td style="text-align: right;">0.486236</td>
<td style="text-align: right;">5.595751</td>
<td style="text-align: right;">12.921917</td>
<td style="text-align: right;">614.7233</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This dataset has 605 observations, with data about different steroidal alkaloids in the fruits of different tomato germplasm grown in 3 locations across 2 years. There is also some other metadata too.</p>
</section>
</section>
<section id="geoms-for-distributions" class="level2">
<h2 class="anchored" data-anchor-id="geoms-for-distributions">Geoms for distributions</h2>
<section id="geom_col" class="level3">
<h3 class="anchored" data-anchor-id="geom_col"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html"><code>geom_col()</code></a></h3>
<p>Often, people use bar charts, representing the height or the length of the bar as proportional to the average value that it represents. These charts are sometimes called dynamite plots because they resemble (when they have an error bar with whisker) those cartoon style dynamite sticks. Pow!</p>
<p>However, these bar charts, even if you add a standard deviation/error, really can hide the true distribution of your data, and for this reason, I and <a href="https://simplystatistics.org/posts/2019-02-21-dynamite-plots-must-die/">others</a> hope you don’t select to make them.</p>
<p>Aside: You may be thinking “Jess you asked us to make one of these in Module 2 homework” and I did but also that was a little different. The plot I asked you to make shows the number of degrees awarded, a value for which there really is no distribution. So in that case we are using a bar plot to show something different than a bar plot which is meant to show somehow an average/median and distribution.</p>
<p>I hope after today, you see that there is always a better chart type to make than a bar chart. But I will show you how to make them anyway.</p>
<p>Before we plot, let’s calculate some summary statistics so we know what we should expect.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Class) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_tomatine =</span> <span class="fu">mean</span>(Tomatine))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 2
  Class                 mean_tomatine
  &lt;chr&gt;                         &lt;dbl&gt;
1 Cultivated Cherry              235.
2 Cultivated Processing          330.
3 S. pimpinellifolium            685.
4 Wide Cross Hybrid              534.
5 Wild Cherry                   4928.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is wrong but an easy mistake to make</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this is not what we want</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-col-wrong-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Just calling <code>geom_col()</code> does not give us what we want. Look at the y-axis scale and how out of line this is with our summary statistics. The reason for this is that <code>geom_col()</code> defaults to <code>position = "stack"</code> which will just sum the alkaloid content across all the observations. Even changing to <code>position = "identity"</code> does not work. This is because we are plotting a transformation of the data (calculation of the mean) which these geoms are not doing.</p>
<p>We can calculate manually by generating the summary values and then piping that into our ggplot call.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Class) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_tomatine =</span> <span class="fu">mean</span>(Tomatine)) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> mean_tomatine)) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-col-manual-summary-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="stat_summary" class="level3">
<h3 class="anchored" data-anchor-id="stat_summary"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html"><code>stat_summary()</code></a></h3>
<p>An easier way to do this would be just with <code>stat_summary()</code>, which does not require the calculation of summary statistic first.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> <span class="st">"mean"</span>, <span class="at">geom =</span> <span class="st">"bar"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-col-stat-summary-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="reordering-x-variables" class="level4">
<h4 class="anchored" data-anchor-id="reordering-x-variables">Reordering x-variables</h4>
<p>Note in these plots the ordering of the x-axis categories – they are alphabetical. This is the ggplot default. There are many reasons why this might not be the most compelling ordering for your data. You may want to order from lowest to highest mean, or in this case, I want to order the tomatoes from most cultivated on the left, to most wild on the right, since this is the prevailing theme of our paper.</p>
<p>We can do this in two ways:</p>
<p>Simply reorder the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set what the order is</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>alkaloids_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cultivated Processing"</span>,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Cultivated Cherry"</span>,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Wide Cross Hybrid"</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Wild Cherry"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"S. pimpinellifolium"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># plot and re-level within aes()</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Class, <span class="at">levels =</span> alkaloids_order), <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> <span class="st">"mean"</span>, <span class="at">geom =</span> <span class="st">"bar"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/reorder-plot-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Change the levels of the data so the reordering happens to every plot in the future.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># what type of variable is Class?</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(alkaloids<span class="sc">$</span>Class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "character"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to factor, and set levels</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>alkaloids<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">factor</span>(alkaloids<span class="sc">$</span>Class,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Cultivated Processing"</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Cultivated Cherry"</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Wide Cross Hybrid"</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Wild Cherry"</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"S. pimpinellifolium"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> <span class="st">"mean"</span>, <span class="at">geom =</span> <span class="st">"bar"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/plot-after-relevelling-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>My tendency would be to re-level the data if I always want to use the same order, and just re-level the plot if I only want to do this once or twice.</p>
</section>
</section>
<section id="geom_boxplot" class="level3">
<h3 class="anchored" data-anchor-id="geom_boxplot"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html"><code>geom_boxplot()</code></a></h3>
<p>A boxplot has the benefit of showing you more than the median and the standard deviation, so you can better see the true distribution of your data. In <code>geom_boxplot()</code>:</p>
<ul>
<li>lower whisker = smallest observation greater than or equal to lower hinge - 1.5 * IQR</li>
<li>lower hinge/bottom line of box part of boxplot = 25% quantile</li>
<li>middle = median, 50% quantile</li>
<li>upper hinge/top line of box part of boxplot = 75% quantile</li>
<li>upper whisker = largest observation less than or equal to upper hinge + 1.5 * IQR</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-boxplot-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>One reason why this is really importantly different from the bar plot is look at the number of outliers we are seeing for Wild Cherry. You don’t capture this at all with the median/mean bar plots.</p>
<p>Because of the scale of this data, it might be beneficial to log transform the y-axis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>) <span class="co"># or scale_y_log10()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3 rows containing non-finite values (`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-boxplot-log-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="geom_jitter" class="level3">
<h3 class="anchored" data-anchor-id="geom_jitter"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html"><code>geom_jitter()</code></a></h3>
<p><code>geom_jitter()</code> is a shortcut for <code>geom_point(position = "jitter")</code>, but is common enough that the shortcut exists. It is often nice to jitter on top of a boxplot. Note, if you don’t want the outliers from <code>geom_boxplot()</code> to be plotted twice, you should indicate <code>outlier.shape = NA</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>) <span class="co"># or scale_y_log10()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis
Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3 rows containing non-finite values (`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-jitter-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Jittering introduces a small amount of variation into your points so they’re easier to see. A width of <code>0</code> is no horizontal jitter. A height of <code>0</code> is no vertical jitter. Typically you don’t want veritcal jitter so that the points retain their fidelity on the y-axis (which is where their concentration is plotted). I basically always use <code>geom_jitter(height = 0)</code> for plots where I want to retain y-axis fidelity.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">height =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>) <span class="co"># or scale_y_log10()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis
Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3 rows containing non-finite values (`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-jitter-height-0-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="geom_histogram" class="level3">
<h3 class="anchored" data-anchor-id="geom_histogram"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html"><code>geom_histogram()</code></a></h3>
<p>We could also look at these distribution more like histograms and it provides to us some additional information. When coupled with faceting, this can be very powerful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">75</span>) <span class="sc">+</span> <span class="co"># default is bins = 30</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>) <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(Class))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/histogram-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(Class))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/density-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="ggridgesgeom_density_ridges" class="level3">
<h3 class="anchored" data-anchor-id="ggridgesgeom_density_ridges"><a href="https://www.rdocumentation.org/packages/ggridges/versions/0.5.2/topics/geom_density_ridges"><code>gg:ridges::geom_density_ridges()</code></a></h3>
<p>I really like the function <code>geom_density_ridges()</code> which is a part of the ggplot add-on package <code>ggridges</code>. It allows you to create ridgeline plots to show distributes in a single non-faceted plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggridges) <span class="co"># for ridgeline plots</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales) <span class="co"># for comma format</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine, <span class="at">y =</span> Class)) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Alpha-tomatine content, µg/100g fresh weight"</span>,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of Alpha-Tomatine Content Across 107 Accessions </span><span class="sc">\n</span><span class="st">of Tomato Grown Across 3 Environments"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-density-ridgeline-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>You can also use the function <a href="https://www.rdocumentation.org/packages/ggridges/versions/0.5.3/topics/stat_density_ridges"><code>geom_density_ridges()</code></a> which will allow you to easily map quantiles or other functions on top of your ridges.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine, <span class="at">y =</span> Class)) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_density_ridges</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">quantile_lines =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">quantiles =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="co"># break into 2 groups, therefore median</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Alpha-tomatine content, µg/100g fresh weight"</span>,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of Alpha-Tomatine Content Across 107 Accessions </span><span class="sc">\n</span><span class="st">of Tomato Grown Across 3 Environments"</span>,</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Black line represents tomato class median concent"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.198</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using the `size` aesthetic with geom_segment was deprecated in ggplot2 3.4.0.
ℹ Please use the `linewidth` aesthetic instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/stat-density-ridgeline-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="changing-class-labels" class="level4">
<h4 class="anchored" data-anchor-id="changing-class-labels">Changing class labels</h4>
<p>I am bothered by the fact that S. pimpinellifolium (a species of wild tomato) is not indicated in italics. We don’t want to italicize all of the <code>labels</code>, just <em>S. pimpinellifolium</em>. Let’s fix that.</p>
<p>We can start by creating a vector of the labels how we want them to appear in the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>class_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cultivated Processing"</span>, </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Cultivated Cherry"</span>,</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Wide Cross Hybrid"</span>, </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Wild Cherry"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">expression</span>(<span class="fu">italic</span>(<span class="st">"S. pimpinellifolium"</span>)))</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>class_labels</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>expression("Cultivated Processing", "Cultivated Cherry", "Wide Cross Hybrid", 
    "Wild Cherry", italic("S. pimpinellifolium"))</code></pre>
</div>
</div>
<p>Then we can use one of the <code>scale_*()</code> functions to change our y-axis scale labels to how we want them to be.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine, <span class="at">y =</span> Class)) <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> class_labels) <span class="sc">+</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Alpha-tomatine content, µg/100g fresh weight"</span>,</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of Alpha-Tomatine Content Across 107 Accessions </span><span class="sc">\n</span><span class="st">of Tomato Grown Across 3 Environments"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.198</code></pre>
</div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/ridgeline-labels-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>If for example your variables were mapped to <code>color</code> or <code>fill</code>, you could do this using <code>scale_color_manual()</code> or <code>scale_fill_manual()</code>, respectively.</p>
</section>
</section>
<section id="ggdist-functions" class="level3">
<h3 class="anchored" data-anchor-id="ggdist-functions"><a href="https://mjskay.github.io/ggdist/index.html"><code>ggdist</code> functions</a></h3>
<p>Another ggplot extension package <code>ggdist</code> has cool geoms you can integrate into ggplots to visualize distributions. I think these work better than <code>geom_dotplot()</code>.</p>
<p>Sometimes using <code>geom_jitter()</code> when you have a lot of data points can look a bit messy. I think in this case, using <code>geom_dots()</code> works very well. The default orientation is is <code>layout = "bin"</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggdist'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ggridges':

    scale_point_color_continuous, scale_point_color_discrete,
    scale_point_colour_continuous, scale_point_colour_discrete,
    scale_point_fill_continuous, scale_point_fill_discrete,
    scale_point_size_continuous</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Tomatine, <span class="at">y =</span> Class)) <span class="sc">+</span>  </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_dots</span>() <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> class_labels) <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Alpha-tomatine, µg/100 g fresh weight"</span>,</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of alkaloid content found among </span><span class="sc">\n</span><span class="st">tomatoes of different classes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/ggdist stat_dots both-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>You can really change the feel of the plot by changing the orientation between horizontal and vertical. If you want to use the orientation <code>layout = "swarm"</code> you need the package <code>ggbeeswarm</code>. This is also a nice package that performs similarly to <code>ggdist</code> but has less functionality which is why I’m covering <code>ggdist</code> here.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm) <span class="co"># required for layout = "swarm"</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_dots</span>(<span class="at">side =</span> <span class="st">"both"</span>, <span class="at">layout =</span> <span class="st">"swarm"</span>) <span class="sc">+</span> <span class="co"># requires ggbeeswarm</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> class_labels) <span class="sc">+</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span> </span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-tomatine, µg/100 g fresh weight"</span>,</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of alkaloid content found among tomatoes of different classes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/ggdist stat_dots both swarm-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>You can also use <code>stat_dotsinterval()</code> which will by default add the median and the interquartile range (though you can change exactly what you want to be displayed).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(<span class="at">side =</span> <span class="st">"both"</span>) <span class="sc">+</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> class_labels) <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span> </span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-tomatine, µg/100 g fresh weight"</span>,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Distribution of alkaloid content found among tomatoes of different classes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/ggdist stat_dotsinterval-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Don’t forget we can keep layering. We can always map other aethetics to our plot (e.g.&nbsp;<code>shape = as.factor(Year)</code>, and we include <code>as.factor()</code> because <code>Year</code> is a character datatype).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Class, <span class="at">y =</span> Tomatine, <span class="at">shape =</span> <span class="fu">as.factor</span>(Year))) <span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">labels =</span> comma) <span class="sc">+</span> </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_dots</span>(<span class="at">side =</span> <span class="st">"both"</span>) <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ggdist</span>() <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">18</span>, .<span class="dv">99</span>),</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="st">"right"</span>, <span class="st">"top"</span>),</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.box.just =</span> <span class="st">"right"</span>,</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.box.background =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.box.margin =</span> <span class="fu">margin</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">shape =</span> <span class="st">"Year"</span>,</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-tomatine (µg/100 g fresh weight)"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="07_distributions_files/figure-html/geom-dots year by shape and legend box-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="useful-resources" class="level2">
<h2 class="anchored" data-anchor-id="useful-resources">Useful resources</h2>
<ul>
<li><p><a href="https://raw.githubusercontent.com/rstudio/cheatsheets/main/data-visualization.pdf"><code>ggplot2</code> cheatsheet</a></p></li>
<li><p><a href="https://ggplot2.tidyverse.org/"><code>ggplot2</code> documentation</a></p></li>
<li><p><a href="https://ggplot2-book.org/index.html">ggplot2: elegant graphics for data analysis by Hadley Wickham</a></p></li>
<li><p><a href="https://github.com/erikgahner/awesome-ggplot2">A really compehensive list of resources compiled by Erik Gahner Larsen</a></p></li>
<li><p><a href="https://wilkelab.org/ggridges/"><code>ggridges</code></a></p></li>
<li><p><a href="https://mjskay.github.io/ggdist/index.html"><code>ggdist</code></a></p></li>
<li><p><a href="https://cran.r-project.org/web/packages/ggbeeswarm/index.html"><code>ggbeeswarm</code></a></p></li>
<li><p>Past ggplot Code Clubs:</p>
<ul>
<li><a href="https://biodash.github.io/codeclub/04_ggplot2/">Visualizing Data by Michael Broe</a></li>
<li><a href="https://biodash.github.io/codeclub/05_ggplot-round-2/">ggplot round 2 by me</a></li>
<li><a href="https://biodash.github.io/codeclub/10_faceting-animating/">Faceting, multi-plots, and animating</a></li>
<li><a href="https://biodash.github.io/codeclub/s02e06_ggplot2/">Visualizing Data by Michael Broe a second one</a></li>
<li><a href="https://biodash.github.io/codeclub/s02e07_ggplot2_part2/">ggplot round 2 a second one by me</a></li>
</ul></li>
</ul>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Created with RStudio and <a href="https://quarto.org/">Quarto</a> <br> © Jessica Cooperstone, 2023, License: <a href="https://creativecommons.org/licenses/by-nc/4.0/">CC BY-NC 4.0</a> <br> <a href="https://osu-codeclub.github.io/#category=website">Want to make a website like this one?</a></div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/jcooperstone/rdataviz">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>