<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jessica Cooperstone">

<title>Visualizing Correlations â€“ ðŸ’»ðŸ§°ðŸ“ŠðŸ¥³</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../img/code-slash.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-647f65b01dad988ac54bf3b04a0860e4.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-06d2be3c2e19c0a3f6003c9d0585ee8c.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-M24LDNCMG6"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-M24LDNCMG6', { 'anonymize_ip': true});
</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../favicon.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">ðŸ’»ðŸ§°ðŸ“ŠðŸ¥³</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-syllabus" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Syllabus</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-syllabus">    
        <li>
    <a class="dropdown-item" href="../../../syllabus/syllabus.html">
 <span class="dropdown-text">Syllabus</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-course-content" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Course content</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-course-content">    
        <li class="dropdown-header">Module 1 - Principles</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module1/01_principles/01_principles.html">
 <span class="dropdown-text">Week 1 - Principles</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module1/02_good-and-bad/02_goodbad.html">
 <span class="dropdown-text">Week 2 - Good and bad visualizations</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd.html">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling.html">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101.html">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102.html">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions.html">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations.html">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats.html">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca.html">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_interactive-plots/11_interactive-plots.html">
 <span class="dropdown-text">Week 11 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_manhattan/12_manhattan.html">
 <span class="dropdown-text">Week 12 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers.html">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-recitations" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Recitations</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-recitations">    
        <li class="dropdown-header">Module 2 - Coding fundamentals</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module2/03_rmarkdown/03_Rmd_recitation.html">
 <span class="dropdown-text">Week 3 - R Markdown</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation.html">
 <span class="dropdown-text">Week 4 - Wrangling</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation.html">
 <span class="dropdown-text">Week 5 - ggplot101</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/06_ggplot102/06_ggplot102_recitation.html">
 <span class="dropdown-text">Week 6 - ggplot102</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 3 - Data exploration</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module3/07_distributions/07_distributions_recitation.html">
 <span class="dropdown-text">Week 7 - Data distributions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/08_correlations/08_correlations_recitation.html">
 <span class="dropdown-text">Week 8 - Correlations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module3/09_add-stats/09_add-stats_recitation.html">
 <span class="dropdown-text">Week 9 - Adding statistics</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Module 4 - Putting it together</li>
        <li>
    <a class="dropdown-item" href="../../../modules/module4/10_pca/10_pca_recitation.html">
 <span class="dropdown-text">Week 10 - Principal components analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/11_interactive-plots/11_interactive-plots_recitation.html">
 <span class="dropdown-text">Week 11 - Interactive plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/12_manhattan/12_manhattan_recitation.html">
 <span class="dropdown-text">Week 12 - Manhattan plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module4/13_leftovers/13_leftovers_recitation.html">
 <span class="dropdown-text">Week 13 - Leftovers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-assignments" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Assignments</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-assignments">    
        <li class="dropdown-header">Module assignments</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module1/module_1_assignment.html">
 <span class="dropdown-text">Module 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module2/module_2_assignment.html">
 <span class="dropdown-text">Module 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module3/module_3_assignment.html">
 <span class="dropdown-text">Module 3</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../assignments/modules/module4/module_4_assignment.html">
 <span class="dropdown-text">Module 4</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Reflections</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/reflections.html">
 <span class="dropdown-text">Reflection instructions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Recitations</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/recitations.html">
 <span class="dropdown-text">Recitation instructions</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Capstone assignment</li>
        <li>
    <a class="dropdown-item" href="../../../assignments/capstone.html">
 <span class="dropdown-text">Capstone instructions</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-solutions" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Solutions</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-solutions">    
        <li class="dropdown-header">Module assignment solutions</li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header">Recitation solutions</li>
        <li>
    <a class="dropdown-item" href="https://raw.githubusercontent.com/jcooperstone/jcooperstone.github.io/main/rmds/03_Rmd_recitation_solutions.Rmd">
 <span class="dropdown-text">Week 3 - R Markdown solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/04_wrangling/04_wrangling_recitation_solutions.html">
 <span class="dropdown-text">Week 4 - Wrangling solutions</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../modules/module2/05_ggplot101/05_ggplot101_recitation_solutions.html">
 <span class="dropdown-text">Week 5 - ggplot101 solutions</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/jcooperstone/rdataviz" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../modules/module3/07_distributions/07_distributions.html">Module 3 - Data exploration</a></li><li class="breadcrumb-item"><a href="../../../modules/module3/08_correlations/08_correlations.html">Week 8 - Correlations</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 1 - Principles</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/01_principles/01_principles.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 1 - Principles</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module1/02_good-and-bad/02_goodbad.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 2 - Good and bad visualizations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 2 - Coding fundamentals</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/03_rmarkdown/03_Rmd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 3 - R Markdown</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/04_wrangling/04_wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 4 - Wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/05_ggplot101/05_ggplot101.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 5 - ggplot101</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module2/06_ggplot102/06_ggplot102.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 6 - ggplot102</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 3 - Data exploration</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/07_distributions/07_distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 7 - Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/08_correlations/08_correlations.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Week 8 - Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module3/09_add-stats/09_add-stats.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 9 - Adding statistics</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Module 4 - Putting it together</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/10_pca/10_pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 10 - Principal components analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/11_interactive-plots/11_interactive-plots.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 11 - Interactive plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/12_manhattan/12_manhattan.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 12 - Manhattan plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../modules/module4/13_leftovers/13_leftovers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Week 13 - Leftovers</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul>
  <li><a href="#load-libraries-and-data" id="toc-load-libraries-and-data" class="nav-link" data-scroll-target="#load-libraries-and-data">Load libraries and data</a></li>
  </ul></li>
  <li><a href="#scatterplots" id="toc-scatterplots" class="nav-link" data-scroll-target="#scatterplots">Scatterplots</a></li>
  <li><a href="#adding-geom_smooth" id="toc-adding-geom_smooth" class="nav-link" data-scroll-target="#adding-geom_smooth">Adding <code>geom_smooth()</code></a>
  <ul>
  <li><a href="#faceted-scatterplots" id="toc-faceted-scatterplots" class="nav-link" data-scroll-target="#faceted-scatterplots">Faceted scatterplots</a></li>
  </ul></li>
  <li><a href="#correlation-matrix-with-cor" id="toc-correlation-matrix-with-cor" class="nav-link" data-scroll-target="#correlation-matrix-with-cor">Correlation matrix with <code>cor()</code></a></li>
  <li><a href="#using-ggcorrplot-from-ggcorrplot" id="toc-using-ggcorrplot-from-ggcorrplot" class="nav-link" data-scroll-target="#using-ggcorrplot-from-ggcorrplot">Using <code>ggcorrplot()</code> from <code>ggcorrplot</code></a></li>
  <li><a href="#using-corrplot-from-corrplot" id="toc-using-corrplot-from-corrplot" class="nav-link" data-scroll-target="#using-corrplot-from-corrplot">Using <code>corrplot()</code> from <code>corrplot</code></a></li>
  <li><a href="#ggally" id="toc-ggally" class="nav-link" data-scroll-target="#ggally"><code>GGally</code></a>
  <ul>
  <li><a href="#ggcorr" id="toc-ggcorr" class="nav-link" data-scroll-target="#ggcorr"><code>ggcorr()</code></a></li>
  <li><a href="#ggpairs" id="toc-ggpairs" class="nav-link" data-scroll-target="#ggpairs"><code>ggpairs()</code></a></li>
  </ul></li>
  <li><a href="#manually-making-correlation-plots-with-reshapemelt-and-ggplot" id="toc-manually-making-correlation-plots-with-reshapemelt-and-ggplot" class="nav-link" data-scroll-target="#manually-making-correlation-plots-with-reshapemelt-and-ggplot">Manually making correlation plots with <code>reshape::melt()</code> and <code>ggplot</code></a>
  <ul>
  <li><a href="#upper-triangle" id="toc-upper-triangle" class="nav-link" data-scroll-target="#upper-triangle">Upper triangle</a></li>
  <li><a href="#lower-triangle" id="toc-lower-triangle" class="nav-link" data-scroll-target="#lower-triangle">Lower triangle</a></li>
  </ul></li>
  <li><a href="#useful-resources" id="toc-useful-resources" class="nav-link" data-scroll-target="#useful-resources">Useful resources</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../modules/module3/07_distributions/07_distributions.html">Module 3 - Data exploration</a></li><li class="breadcrumb-item"><a href="../../../modules/module3/08_correlations/08_correlations.html">Week 8 - Correlations</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Visualizing Correlations</h1>
<p class="subtitle lead">Week 8</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jessica Cooperstone </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://imgs.xkcd.com/comics/correlation.png" class="img-fluid figure-img" style="width:70.0%" alt="A cartoon about how correlation is not causation"></p>
<figcaption>Figure from <a href="https://xkcd.com/552/">XKCD</a></figcaption>
</figure>
</div>
</div>
</div>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>We will will building on our lessons on ggplot101 and ggplot102 which focused on an overall understanding of the grammar of graphics, basic syntax, adding data, aesthetic mappings, geoms, facets, scales, labels, and themes. Today we are going to apply what we learned towards trying to better understanding and visualize correlations within our data. To do this we will also use some ggplot extension packages.</p>
<section id="load-libraries-and-data" class="level3">
<h3 class="anchored" data-anchor-id="load-libraries-and-data">Load libraries and data</h3>
<p>Before we get started, letâ€™s load our libraries.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Today we are going to continue to use the same real research data from my group from last week. We will be reading in the supplementary data from a <a href="https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/tpg2.20192">paper</a> written by Michael Dzakovich, and published in The Plant Genome. The data is present in a Excel worksheet, so we will use the function <code>read_excel()</code> from the tidyverse (but not core tidyverse) package <code>readxl</code>. We want to import Supplemental Table 1. You can indicate which sheet you want to import in the arguments to <code>read_excel()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">"tpg220192-sup-0002-supmat.xlsx"</span>,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">sheet =</span> <span class="st">"S1 Raw Data Diversity Panel"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(alkaloids))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">ID</th>
<th style="text-align: right;">Year</th>
<th style="text-align: left;">Environment</th>
<th style="text-align: right;">Block</th>
<th style="text-align: left;">Genotype</th>
<th style="text-align: left;">Plot_Source</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Origin</th>
<th style="text-align: left;">Provence</th>
<th style="text-align: left;">Blanca_Cluster1</th>
<th style="text-align: left;">Blanca_Cluster2</th>
<th style="text-align: left;">Passport_Species</th>
<th style="text-align: left;">Passport_Classification</th>
<th style="text-align: left;">Sim_Grouping</th>
<th style="text-align: left;">Latitude</th>
<th style="text-align: left;">Longitude</th>
<th style="text-align: right;">Dehydrotomatidine</th>
<th style="text-align: right;">Tomatidine</th>
<th style="text-align: right;">Dehydrotomatine1</th>
<th style="text-align: right;">Dehydrotomatine2</th>
<th style="text-align: right;">TotalDehydrotomatine</th>
<th style="text-align: right;">Tomatine</th>
<th style="text-align: right;">Hydroxytomatine1</th>
<th style="text-align: right;">Hydroxytomatine2</th>
<th style="text-align: right;">Hydroxytomatine3</th>
<th style="text-align: right;">Hydroxytomatine4</th>
<th style="text-align: right;">TotalHydroxytomatine</th>
<th style="text-align: right;">Acetoxytomatine1</th>
<th style="text-align: right;">Acetoxytomatine2</th>
<th style="text-align: right;">Acetoxytomatine3</th>
<th style="text-align: right;">TotalAcetoxytomatine</th>
<th style="text-align: right;">DehydrolycoperosideFGdehydroesculeosideA</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA1</th>
<th style="text-align: right;">LycoperosideFGEsculeosideA2</th>
<th style="text-align: right;">TotalLycoperosideFGEsculeosideA</th>
<th style="text-align: right;">EsculeosideB1</th>
<th style="text-align: right;">EsculeosideB2</th>
<th style="text-align: right;">EsculeosideB3</th>
<th style="text-align: right;">TotalEsculeosideB</th>
<th style="text-align: right;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">7805</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreEarly18</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">5.726010</td>
<td style="text-align: right;">0.350331</td>
<td style="text-align: right;">6.076341</td>
<td style="text-align: right;">172.66244</td>
<td style="text-align: right;">1.079190</td>
<td style="text-align: right;">86.72742</td>
<td style="text-align: right;">17.831892</td>
<td style="text-align: right;">9.142607</td>
<td style="text-align: right;">114.78111</td>
<td style="text-align: right;">18.902399</td>
<td style="text-align: right;">56.307182</td>
<td style="text-align: right;">1.890053</td>
<td style="text-align: right;">77.099634</td>
<td style="text-align: right;">5.125904</td>
<td style="text-align: right;">10.277325</td>
<td style="text-align: right;">336.8893</td>
<td style="text-align: right;">347.1666</td>
<td style="text-align: right;">3.787979</td>
<td style="text-align: right;">0.924195</td>
<td style="text-align: right;">3.943230</td>
<td style="text-align: right;">8.655404</td>
<td style="text-align: right;">731.5675</td>
</tr>
<tr class="even">
<td style="text-align: right;">7898</td>
<td style="text-align: right;">2017</td>
<td style="text-align: left;">Fre17</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K9-8584</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.169068</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">55.47329</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">53.32292</td>
<td style="text-align: right;">13.630697</td>
<td style="text-align: right;">4.841762</td>
<td style="text-align: right;">71.79538</td>
<td style="text-align: right;">3.557348</td>
<td style="text-align: right;">4.107289</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">7.664637</td>
<td style="text-align: right;">2.905500</td>
<td style="text-align: right;">5.548102</td>
<td style="text-align: right;">199.6694</td>
<td style="text-align: right;">205.2175</td>
<td style="text-align: right;">8.978931</td>
<td style="text-align: right;">1.897850</td>
<td style="text-align: right;">6.794690</td>
<td style="text-align: right;">17.671471</td>
<td style="text-align: right;">360.8969</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7523</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreLate18</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.135675</td>
<td style="text-align: right;">0.680554</td>
<td style="text-align: right;">5.073552</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">5.073552</td>
<td style="text-align: right;">123.85835</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">50.90989</td>
<td style="text-align: right;">6.503939</td>
<td style="text-align: right;">1.368847</td>
<td style="text-align: right;">58.78268</td>
<td style="text-align: right;">3.931461</td>
<td style="text-align: right;">4.123222</td>
<td style="text-align: right;">0.623340</td>
<td style="text-align: right;">8.678023</td>
<td style="text-align: right;">2.185082</td>
<td style="text-align: right;">5.104115</td>
<td style="text-align: right;">259.0177</td>
<td style="text-align: right;">264.1218</td>
<td style="text-align: right;">4.049145</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">6.749386</td>
<td style="text-align: right;">10.798531</td>
<td style="text-align: right;">474.3143</td>
</tr>
<tr class="even">
<td style="text-align: right;">7724</td>
<td style="text-align: right;">2017</td>
<td style="text-align: left;">Fre17</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K9-8584</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.054300</td>
<td style="text-align: right;">0.497261</td>
<td style="text-align: right;">19.419087</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">19.419087</td>
<td style="text-align: right;">239.01264</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">36.02318</td>
<td style="text-align: right;">8.557673</td>
<td style="text-align: right;">7.483933</td>
<td style="text-align: right;">52.06478</td>
<td style="text-align: right;">3.341048</td>
<td style="text-align: right;">16.415426</td>
<td style="text-align: right;">1.057100</td>
<td style="text-align: right;">20.813574</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">203.0061</td>
<td style="text-align: right;">203.0061</td>
<td style="text-align: right;">1.678210</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">2.349633</td>
<td style="text-align: right;">4.027843</td>
<td style="text-align: right;">538.8955</td>
</tr>
<tr class="odd">
<td style="text-align: right;">7427</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreLate18</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.139454</td>
<td style="text-align: right;">0.553801</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">64.31783</td>
<td style="text-align: right;">0.879435</td>
<td style="text-align: right;">39.91027</td>
<td style="text-align: right;">7.228388</td>
<td style="text-align: right;">3.015298</td>
<td style="text-align: right;">51.03339</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.131685</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.131685</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">4.054211</td>
<td style="text-align: right;">299.5687</td>
<td style="text-align: right;">303.6229</td>
<td style="text-align: right;">10.146857</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">4.882339</td>
<td style="text-align: right;">15.029197</td>
<td style="text-align: right;">437.8283</td>
</tr>
<tr class="even">
<td style="text-align: right;">7854</td>
<td style="text-align: right;">2018</td>
<td style="text-align: left;">FreEarly18</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">CULBPT_05_11</td>
<td style="text-align: left;">2K17-7724</td>
<td style="text-align: left;">Cultivated Processing</td>
<td style="text-align: left;">USA</td>
<td style="text-align: left;">NY</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL_processing_2</td>
<td style="text-align: left;">SLL</td>
<td style="text-align: left;">SLL_processing_NY</td>
<td style="text-align: left;">Arid</td>
<td style="text-align: left;">40.712800000000001</td>
<td style="text-align: left;">-74.006</td>
<td style="text-align: right;">0.049700</td>
<td style="text-align: right;">0.262174</td>
<td style="text-align: right;">3.737579</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">3.737579</td>
<td style="text-align: right;">68.44913</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">23.86864</td>
<td style="text-align: right;">13.506299</td>
<td style="text-align: right;">1.456982</td>
<td style="text-align: right;">38.83192</td>
<td style="text-align: right;">4.657902</td>
<td style="text-align: right;">4.259007</td>
<td style="text-align: right;">0.605729</td>
<td style="text-align: right;">9.522638</td>
<td style="text-align: right;">9.832149</td>
<td style="text-align: right;">11.595595</td>
<td style="text-align: right;">459.5205</td>
<td style="text-align: right;">471.1161</td>
<td style="text-align: right;">6.839930</td>
<td style="text-align: right;">0.486236</td>
<td style="text-align: right;">5.595751</td>
<td style="text-align: right;">12.921917</td>
<td style="text-align: right;">614.7233</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This dataset has 605 observations, with data about different steroidal alkaloids in the fruits of different tomato germplasm grown in 3 locations across 2 years. There is also some other metadata too.</p>
<p>For those who are chemistry minded, here is a little pathway context for the compounds we are investigating today.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/esm/art%3A10.1038%2Fs41588-020-0690-6/MediaObjects/41588_2020_690_Fig12_ESM.jpg?as=webp" class="img-fluid figure-img"></p>
<figcaption>Figure from <a href="https://www.nature.com/articles/s41588-020-0690-6">SyzmaÅ„sky et al., Nature Genetics 2020</a></figcaption>
</figure>
</div>
</section>
</section>
<section id="scatterplots" class="level2">
<h2 class="anchored" data-anchor-id="scatterplots">Scatterplots</h2>
<p>A very simple first pass way to understand if you have relationships within your data is to make scatterplots of the variables you think might be correlated. Letâ€™s start by investigating how the different alkaloid concentrations are correlated to each other. First we will see how alpha-tomatine content (<code>Tomatine</code>) is related to total steroidal alkaloid content (<code>Total</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Total, <span class="at">y =</span> Tomatine)) <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Total Steroidal Alkaloids (Âµg/100 g)"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-Tomatine (Âµg/100 g)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/alphatomatine vs total alkaloids-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>It seems like there are two separate groups here - the points with a steeper slope, and the points with a less steep slope. We can color our points based on what <code>Class</code> of tomato the data comes from, maybe that will reveal something. In the meanwhile letâ€™s make this plot look a bit nicer. The package <a href="https://www.rdocumentation.org/packages/scales/versions/0.4.1"><code>scales</code></a> has some nice functions that help you control the scaling of your plots, in this case, making each of the axes have numbers in <a href="https://www.rdocumentation.org/packages/scales/versions/0.4.1/topics/comma_format"><code>comma_format()</code></a>. I also am using the hex codes for a color-blind friendly qualitative color scheme developed by <a href="https://personal.sron.nl/~pault/#sec:qualitative">Paul Tol</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Total, <span class="at">y =</span> Tomatine, <span class="at">color =</span> Class)) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#4477AA"</span>, <span class="st">"#EE6677"</span>, <span class="st">"#228833"</span>, <span class="st">"#CCBB44"</span>, <span class="st">"#66CCEE"</span>)) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Total Steroidal Alkaloids (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-Tomatine (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Relationship between Alpha-Tomatine and Total Steroidal Alkaloids </span><span class="sc">\n</span><span class="st">Across Different Germplasm in the Red Tomato Clade"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/alphatomatine vs total alkaloids color-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>All of the tomatoes in the two extremes of this plot are from the <code>Class</code> Wild Cherry. What would this look like if we removed these fruits? Note, I adjusted the color scale to remove the hex code associated with Wild Cherry but keeping the other colors the same.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Class <span class="sc">!=</span> <span class="st">"Wild Cherry"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Total, <span class="at">y =</span> Tomatine, <span class="at">color =</span> Class)) <span class="sc">+</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#4477AA"</span>, <span class="st">"#EE6677"</span>, <span class="st">"#228833"</span>, <span class="st">"#CCBB44"</span>)) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Total Steroidal Alkaloids (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-Tomatine (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Relationship between Alpha-Tomatine and Total Steroidal Alkaloids </span><span class="sc">\n</span><span class="st">Across Different Germplasm in the Red Tomato Clade"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/alphatomatine vs total no wild cherry-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="adding-geom_smooth" class="level2">
<h2 class="anchored" data-anchor-id="adding-geom_smooth">Adding <code>geom_smooth()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Total, <span class="at">y =</span> Tomatine, <span class="at">color =</span> Class)) <span class="sc">+</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">comma_format</span>(<span class="at">big.mark =</span> <span class="st">","</span>)) <span class="sc">+</span> <span class="co"># requires the package scales</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#4477AA"</span>, <span class="st">"#EE6677"</span>, <span class="st">"#228833"</span>, <span class="st">"#CCBB44"</span>, <span class="st">"#66CCEE"</span>)) <span class="sc">+</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Total Steroidal Alkaloids (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-Tomatine (Âµg/100 g fresh weight)"</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Relationship between Alpha-Tomatine and Total Steroidal Alkaloids </span><span class="sc">\n</span><span class="st">Across Different Germplasm in the Red Tomato Clade"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/adding geom smooth-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="faceted-scatterplots" class="level3">
<h3 class="anchored" data-anchor-id="faceted-scatterplots">Faceted scatterplots</h3>
<p>We may be able to see trends by tomato class more easily if we facet our scatterplots. I also am demonstrating here how within the ggplot function you can make alter the aesthetics you plot - here I am turning data that is present as Âµg/100 g to mg/100 g by dividing by 1000 and changing the axis labels accordingly.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Total<span class="sc">/</span><span class="dv">1000</span>, <span class="at">y =</span> Tomatine<span class="sc">/</span><span class="dv">1000</span>, <span class="at">color =</span> Class)) <span class="sc">+</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#4477AA"</span>, <span class="st">"#EE6677"</span>, <span class="st">"#228833"</span>, <span class="st">"#CCBB44"</span>, <span class="st">"#66CCEE"</span>)) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(Class), <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Total Steroidal Alkaloids (mg/100 g fresh weight)"</span>,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Alpha-Tomatine (mg/100 g fresh weight)"</span>,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Relationship between Alpha-Tomatine and Total Steroidal Alkaloids </span><span class="sc">\n</span><span class="st">Across Different Germplasm in the Red Tomato Clade"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/faceted sc atterplots-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="correlation-matrix-with-cor" class="level2">
<h2 class="anchored" data-anchor-id="correlation-matrix-with-cor">Correlation matrix with <code>cor()</code></h2>
<p><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a> is a function from base R that will allow you to create a correlation matrix.</p>
<p>Before we use <code>cor()</code> we will clean up our dataset to include only the variables we want to correlate.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(alkaloids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                                      
 [2] "Year"                                    
 [3] "Environment"                             
 [4] "Block"                                   
 [5] "Genotype"                                
 [6] "Plot_Source"                             
 [7] "Class"                                   
 [8] "Origin"                                  
 [9] "Provence"                                
[10] "Blanca_Cluster1"                         
[11] "Blanca_Cluster2"                         
[12] "Passport_Species"                        
[13] "Passport_Classification"                 
[14] "Sim_Grouping"                            
[15] "Latitude"                                
[16] "Longitude"                               
[17] "Dehydrotomatidine"                       
[18] "Tomatidine"                              
[19] "Dehydrotomatine1"                        
[20] "Dehydrotomatine2"                        
[21] "TotalDehydrotomatine"                    
[22] "Tomatine"                                
[23] "Hydroxytomatine1"                        
[24] "Hydroxytomatine2"                        
[25] "Hydroxytomatine3"                        
[26] "Hydroxytomatine4"                        
[27] "TotalHydroxytomatine"                    
[28] "Acetoxytomatine1"                        
[29] "Acetoxytomatine2"                        
[30] "Acetoxytomatine3"                        
[31] "TotalAcetoxytomatine"                    
[32] "DehydrolycoperosideFGdehydroesculeosideA"
[33] "LycoperosideFGEsculeosideA1"             
[34] "LycoperosideFGEsculeosideA2"             
[35] "TotalLycoperosideFGEsculeosideA"         
[36] "EsculeosideB1"                           
[37] "EsculeosideB2"                           
[38] "EsculeosideB3"                           
[39] "TotalEsculeosideB"                       
[40] "Total"                                   </code></pre>
</div>
</div>
<p>From looking at the colnames and reading the supplemental information, we can see that some columns are composites of others. For example, the column <code>TotalAcetoxytomatine</code> = <code>Acetoxytomatine1</code> + <code>Acetoxytomatine2</code> + <code>Acetoxytomatine3</code>. So we want to pull only the columns that represent the total for any given alkaloids. There should be 10 columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a vector of the names we want to keep</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>alkaloid_total_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Dehydrotomatidine"</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Tomatidine"</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalDehydrotomatine"</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Tomatine"</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalHydroxytomatine"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalAcetoxytomatine"</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"DehydrolycoperosideFGdehydroesculeosideA"</span>,</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalLycoperosideFGEsculeosideA"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"TotalEsculeosideB"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"Total"</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co"># make a new df including some metadata and the alkaloid_total_names</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>alkaloids_totals <span class="ot">&lt;-</span> alkaloids <span class="sc">%&gt;%</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID, Year, Environment, Block, Genotype, Class, <span class="fu">all_of</span>(alkaloid_total_names))</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co"># did it work? look at colnames()</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(alkaloids_totals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                                      
 [2] "Year"                                    
 [3] "Environment"                             
 [4] "Block"                                   
 [5] "Genotype"                                
 [6] "Class"                                   
 [7] "Dehydrotomatidine"                       
 [8] "Tomatidine"                              
 [9] "TotalDehydrotomatine"                    
[10] "Tomatine"                                
[11] "TotalHydroxytomatine"                    
[12] "TotalAcetoxytomatine"                    
[13] "DehydrolycoperosideFGdehydroesculeosideA"
[14] "TotalLycoperosideFGEsculeosideA"         
[15] "TotalEsculeosideB"                       
[16] "Total"                                   </code></pre>
</div>
</div>
<p>Now we can create a correlation matrix to see how each of our 10 alkaloids is correlated to the concentration of each other alkaloid (including the compile metric of <code>Total</code> which sums all the alkaloids). The default for <code>cor()</code> is to use Pearsonâ€™s correlation coefficient, but you can set to use Spearman <code>method = "spearman"</code> or Kendall <code>method = "kendall"</code> if you prefer. Check the documentation for <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a> for more information.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>alkaloids_cor <span class="ot">&lt;-</span> alkaloids_totals <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(alkaloid_total_names)) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor</span>()</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># look at our correlation matrix</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(alkaloids_cor) <span class="co"># kable makes a nicely formatted table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 16%">
<col style="width: 7%">
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 16%">
<col style="width: 13%">
<col style="width: 7%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Dehydrotomatidine</th>
<th style="text-align: right;">Tomatidine</th>
<th style="text-align: right;">TotalDehydrotomatine</th>
<th style="text-align: right;">Tomatine</th>
<th style="text-align: right;">TotalHydroxytomatine</th>
<th style="text-align: right;">TotalAcetoxytomatine</th>
<th style="text-align: right;">DehydrolycoperosideFGdehydroesculeosideA</th>
<th style="text-align: right;">TotalLycoperosideFGEsculeosideA</th>
<th style="text-align: right;">TotalEsculeosideB</th>
<th style="text-align: right;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dehydrotomatidine</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.2974462</td>
<td style="text-align: right;">0.0324918</td>
<td style="text-align: right;">0.0238230</td>
<td style="text-align: right;">0.0099126</td>
<td style="text-align: right;">0.0322029</td>
<td style="text-align: right;">0.0305049</td>
<td style="text-align: right;">0.0761907</td>
<td style="text-align: right;">0.0282219</td>
<td style="text-align: right;">0.0708252</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tomatidine</td>
<td style="text-align: right;">0.2974462</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.3744672</td>
<td style="text-align: right;">0.3736949</td>
<td style="text-align: right;">0.1003558</td>
<td style="text-align: right;">0.0382981</td>
<td style="text-align: right;">-0.0059964</td>
<td style="text-align: right;">0.0373649</td>
<td style="text-align: right;">0.0126724</td>
<td style="text-align: right;">0.2044979</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TotalDehydrotomatine</td>
<td style="text-align: right;">0.0324918</td>
<td style="text-align: right;">0.3744672</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.9214859</td>
<td style="text-align: right;">0.2290192</td>
<td style="text-align: right;">0.4011257</td>
<td style="text-align: right;">-0.0820469</td>
<td style="text-align: right;">-0.1149682</td>
<td style="text-align: right;">-0.1217560</td>
<td style="text-align: right;">0.5636969</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tomatine</td>
<td style="text-align: right;">0.0238230</td>
<td style="text-align: right;">0.3736949</td>
<td style="text-align: right;">0.9214859</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.0995212</td>
<td style="text-align: right;">0.1220596</td>
<td style="text-align: right;">-0.1140360</td>
<td style="text-align: right;">-0.1357819</td>
<td style="text-align: right;">-0.1260377</td>
<td style="text-align: right;">0.3756155</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TotalHydroxytomatine</td>
<td style="text-align: right;">0.0099126</td>
<td style="text-align: right;">0.1003558</td>
<td style="text-align: right;">0.2290192</td>
<td style="text-align: right;">0.0995212</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.3563506</td>
<td style="text-align: right;">0.0330078</td>
<td style="text-align: right;">0.0284887</td>
<td style="text-align: right;">0.0134806</td>
<td style="text-align: right;">0.4774036</td>
</tr>
<tr class="even">
<td style="text-align: left;">TotalAcetoxytomatine</td>
<td style="text-align: right;">0.0322029</td>
<td style="text-align: right;">0.0382981</td>
<td style="text-align: right;">0.4011257</td>
<td style="text-align: right;">0.1220596</td>
<td style="text-align: right;">0.3563506</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">-0.0865506</td>
<td style="text-align: right;">-0.1106212</td>
<td style="text-align: right;">-0.0947254</td>
<td style="text-align: right;">0.6782337</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DehydrolycoperosideFGdehydroesculeosideA</td>
<td style="text-align: right;">0.0305049</td>
<td style="text-align: right;">-0.0059964</td>
<td style="text-align: right;">-0.0820469</td>
<td style="text-align: right;">-0.1140360</td>
<td style="text-align: right;">0.0330078</td>
<td style="text-align: right;">-0.0865506</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.8862982</td>
<td style="text-align: right;">0.7401116</td>
<td style="text-align: right;">0.4792108</td>
</tr>
<tr class="even">
<td style="text-align: left;">TotalLycoperosideFGEsculeosideA</td>
<td style="text-align: right;">0.0761907</td>
<td style="text-align: right;">0.0373649</td>
<td style="text-align: right;">-0.1149682</td>
<td style="text-align: right;">-0.1357819</td>
<td style="text-align: right;">0.0284887</td>
<td style="text-align: right;">-0.1106212</td>
<td style="text-align: right;">0.8862982</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.7779405</td>
<td style="text-align: right;">0.5222751</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TotalEsculeosideB</td>
<td style="text-align: right;">0.0282219</td>
<td style="text-align: right;">0.0126724</td>
<td style="text-align: right;">-0.1217560</td>
<td style="text-align: right;">-0.1260377</td>
<td style="text-align: right;">0.0134806</td>
<td style="text-align: right;">-0.0947254</td>
<td style="text-align: right;">0.7401116</td>
<td style="text-align: right;">0.7779405</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">0.4059824</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">0.0708252</td>
<td style="text-align: right;">0.2044979</td>
<td style="text-align: right;">0.5636969</td>
<td style="text-align: right;">0.3756155</td>
<td style="text-align: right;">0.4774036</td>
<td style="text-align: right;">0.6782337</td>
<td style="text-align: right;">0.4792108</td>
<td style="text-align: right;">0.5222751</td>
<td style="text-align: right;">0.4059824</td>
<td style="text-align: right;">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note the diagonal is all composed of 1s. This makes sense because the correlation of each alkaloid with itself is 1.</p>
</section>
<section id="using-ggcorrplot-from-ggcorrplot" class="level2">
<h2 class="anchored" data-anchor-id="using-ggcorrplot-from-ggcorrplot">Using <code>ggcorrplot()</code> from <code>ggcorrplot</code></h2>
<p>Use the function <code>ggcorrplot()</code> without any additional arguments besides the correlation matrix <code>alkaloids_cor</code>. In general, I think if you want to make a bunch of correlation plots quickly, and donâ€™t intend to publish them, `<code>ggcorrplot()</code> works well, but the visuals of the plot are quite difficult to customize.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcorrplot</span>(alkaloids_cor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggcorrplot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is not a perfect plot but its a good starting point. Correlation matrices are inherently symmetric, meaning if we display only the top or bottom triangle, we do not lose any information. We will work on editing this plot in different ways to show more information and make it more beautiful.</p>
<p>We could also make the plot circles instead of squares at the same time.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcorrplot</span>(alkaloids_cor, </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">method =</span> <span class="st">"circle"</span>,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">type =</span> <span class="st">"lower"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggcorrplot lower circle-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In general, I think if you want to make a bunch of correlation plots quickly, and donâ€™t intend to publish them, `<code>ggcorrplot()</code> works well, but the visuals of the plot are quite difficult to customize.</p>
</section>
<section id="using-corrplot-from-corrplot" class="level2">
<h2 class="anchored" data-anchor-id="using-corrplot-from-corrplot">Using <code>corrplot()</code> from <code>corrplot</code></h2>
<p>Similarly, you can use a base R plotting based package <code>corrplot()</code> to make correlation plots. The customization syntax here is quite different from what weâ€™ve been working with in ggplot, but I wanted you to feel familiar with some base R tools.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>corrplot 0.95 loaded</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(alkaloids_cor, <span class="at">type =</span> <span class="st">"lower"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/corrplot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>I have used <code>corrplot()</code> in publications before and felt like I couldnâ€™t customize the plots as much as I wanted. In the process of putting together this content, I learned some news ways to customize these plots that are actually very nice. Here are some parameters you can modify in R. You can also order your variables by hierarchical clustering.</p>
<p>First we will start (as we always do) by wrangling.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create matrix for correlation</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>alkaloids_to_cor <span class="ot">&lt;-</span> alkaloids_totals <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(alkaloid_total_names)) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="co"># rcorr() needs a matrix</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Hmisc) <span class="co"># does cor() but also computes significance levels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Hmisc'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    src, summarize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    format.pval, units</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a matrix of pvalues for the correlations</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>alkaloids_rcorr <span class="ot">=</span> <span class="fu">rcorr</span>(alkaloids_to_cor, <span class="at">type =</span> <span class="st">"pearson"</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># create a vector of the alkaloid names for labeling</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>alkaloid_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Dehydrotomatidine"</span>,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Tomatidine"</span>,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Dehydrotomatine"</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Alpha-Tomatine"</span>,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Hydroxytomatine"</span>,</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Acetoxytomatine"</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Dehydrlycoperoside F, G, </span><span class="sc">\n</span><span class="st">or Dehydroescueloside A"</span>,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lycoperoside F, G, </span><span class="sc">\n</span><span class="st">or Escueloside A"</span>,</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Escueloside B"</span>,</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Total Steroidal Alkaloids"</span>)</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="co"># change row and column names of the correlation matrix</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co"># so they are how we want them to be plotted</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(alkaloids_rcorr<span class="sc">$</span>r) <span class="ot">&lt;-</span> alkaloid_labels</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(alkaloids_rcorr<span class="sc">$</span>r) <span class="ot">&lt;-</span> alkaloid_labels</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co"># change row and column names of the pvalue matrix</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co"># so they are how we want them to be plotted</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(alkaloids_rcorr<span class="sc">$</span>P) <span class="ot">&lt;-</span> alkaloid_labels</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(alkaloids_rcorr<span class="sc">$</span>P) <span class="ot">&lt;-</span> alkaloid_labels</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we are ready to plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(alkaloids_rcorr<span class="sc">$</span>r, <span class="co"># the correlation matrix</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">type =</span> <span class="st">"lower"</span>, <span class="co"># lower triangle</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">tl.col =</span> <span class="st">"black"</span>, <span class="co"># axis labels are black</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">p.mat  =</span> alkaloids_rcorr<span class="sc">$</span>P, <span class="co"># pvalue matrix</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">sig.level =</span> <span class="fl">0.05</span>, <span class="co"># how sig does a cor need to be to be included</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">insig =</span> <span class="st">"blank"</span>, <span class="co"># do not display insignificant correlations</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">addCoef.col =</span> <span class="st">"black"</span>, <span class="co"># display correlations in black</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">diag =</span> <span class="cn">FALSE</span>, <span class="co"># don't show the diagonal (because this is all 1)</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">number.cex =</span> <span class="fl">0.6</span>) <span class="co"># size of correlation font</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/rcorr customized-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ggally" class="level2">
<h2 class="anchored" data-anchor-id="ggally"><code>GGally</code></h2>
<section id="ggcorr" class="level3">
<h3 class="anchored" data-anchor-id="ggcorr"><code>ggcorr()</code></h3>
<p>Another ggplot extension package <code>ggally</code> has the function <a href="https://briatte.github.io/ggcorr/"><code>ggcorr()</code></a> which also allows the creation of correlation plots, but ones that are more easily customizable. <code>ggcorr()</code> objects are moderately customizable. They make work for some of you so Iâ€™m sharing how to make them.</p>
<p>Note, <code>GGally::ggcorr()</code> does not take a correlation matrix, but instead takes the data you want to make a correlation matrix for. You can specific the method of correlation in the arguments. The default is Pearsonâ€™s correlation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>to_corr <span class="ot">&lt;-</span> alkaloids_totals <span class="sc">%&gt;%</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(alkaloid_total_names)) </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcorr</span>(to_corr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggcorr-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There is only one labeled axis - this is because there is no diagonal in these plots, like we saw with <code>ggcorrplot()</code> and <code>corrplot()</code>.</p>
<p>We can now spend some time improving the aesthetics of our plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcorr</span>(to_corr, <span class="co"># data for correlation</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">low =</span> <span class="st">"#f1a340"</span>, <span class="co"># -1 correlation color</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">mid =</span> <span class="st">"#f7f7f7"</span>, <span class="co"># 0 correlation color</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">high =</span> <span class="st">"#998ec3"</span>) <span class="co"># 1 correlation color</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggcorr customized-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcorr</span>(to_corr,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">low =</span> <span class="st">"#f1a340"</span>, <span class="at">mid =</span> <span class="st">"#f7f7f7"</span>, <span class="at">high =</span> <span class="st">"#998ec3"</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">geom =</span> <span class="st">"circle"</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">label_size =</span> <span class="dv">2</span>, </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">label_round =</span> <span class="dv">2</span>,</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">layout.exp =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggcorr customized-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>For this example, we have very long label names which are really difficult to wrap, but if your labels are more reasonable this may work well for you.</p>
</section>
<section id="ggpairs" class="level3">
<h3 class="anchored" data-anchor-id="ggpairs"><code>ggpairs()</code></h3>
<p>We can also use the function <code>GGally::ggpairs()</code> to make a matrix of correlation related plots.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>alkaloids_totals <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggpairs</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"Tomatine"</span>, <span class="st">"TotalLycoperosideFGEsculeosideA"</span>, <span class="st">"Total"</span>), <span class="co"># pick variables</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">aes</span>(<span class="at">color =</span> Class))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggpairs-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Letâ€™s customized a bit.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove zeroes since they don't log transform</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co"># make log transformed columns</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>alkaloids_totals_log <span class="ot">&lt;-</span> alkaloids_totals <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tomatine <span class="sc">!=</span> <span class="dv">0</span>, </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>         TotalLycoperosideFGEsculeosideA <span class="sc">!=</span> <span class="dv">0</span>,</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>         Total <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10_tomatine =</span> <span class="fu">log10</span>(Tomatine),</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">log10_FGA =</span> <span class="fu">log10</span>(TotalLycoperosideFGEsculeosideA),</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">log10_total =</span> <span class="fu">log10</span>(Total))</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>alkaloids_totals_log <span class="sc">%&gt;%</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggpairs</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"log10_tomatine"</span>, <span class="st">"log10_FGA"</span>, <span class="st">"log10_total"</span>),</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>          <span class="fu">aes</span>(<span class="at">color =</span> Class, <span class="at">alpha =</span> <span class="fl">0.5</span>), <span class="co"># note alpha inside aes which is weird idk why</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">columnLabels =</span> <span class="fu">c</span>(<span class="st">"Alpha-Tomatine"</span>, <span class="st">"Lycoperoside F/G</span><span class="sc">\n</span><span class="st"> Escueloside A"</span>, <span class="st">"Total Alkaloids"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/ggpairs customized-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="manually-making-correlation-plots-with-reshapemelt-and-ggplot" class="level2">
<h2 class="anchored" data-anchor-id="manually-making-correlation-plots-with-reshapemelt-and-ggplot">Manually making correlation plots with <code>reshape::melt()</code> and <code>ggplot</code></h2>
<p>Because some of the correlation specific packages are hard to customize, I am going to show you how to make your own plots by reshaping your data with <code>reshape2::melt()</code> and some base R functions, and plotting using the standard ggplot syntax.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2) <span class="co"># contains melt()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'reshape2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    smiths</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># take cor matrix and convert to df with 3 columns: Var1, Var2, and value</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>melted_alkaloids_cor <span class="ot">&lt;-</span> <span class="fu">melt</span>(alkaloids_cor)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># what does it look like?</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(melted_alkaloids_cor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  Var1              Var2       value
1    Dehydrotomatidine Dehydrotomatidine 1.000000000
2           Tomatidine Dehydrotomatidine 0.297446153
3 TotalDehydrotomatine Dehydrotomatidine 0.032491778
4             Tomatine Dehydrotomatidine 0.023823011
5 TotalHydroxytomatine Dehydrotomatidine 0.009912624
6 TotalAcetoxytomatine Dehydrotomatidine 0.032202892</code></pre>
</div>
</div>
<p>First pass minimalist plotting</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>melted_alkaloids_cor <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/manual plotting 1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Lots to fix! What if we want only the upper or lower triangle, again since this plot is symmetric.</p>
<section id="upper-triangle" class="level3">
<h3 class="anchored" data-anchor-id="upper-triangle">Upper triangle</h3>
<p>Keep only the upper triangle.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># "save as"</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>alkaloids_upper <span class="ot">&lt;-</span> alkaloids_cor</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># use function lower.tri() and set the lower triangle all to NA</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co"># then we can keep only the upper triangle</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>alkaloids_upper[<span class="fu">lower.tri</span>(alkaloids_upper)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="co"># melt to go back to long format</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>melted_alkaloids_upper <span class="ot">&lt;-</span> <span class="fu">melt</span>(alkaloids_upper, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co"># did it work?</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(melted_alkaloids_upper) <span class="co"># yup</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   Var1                 Var2      value
1     Dehydrotomatidine    Dehydrotomatidine 1.00000000
11    Dehydrotomatidine           Tomatidine 0.29744615
12           Tomatidine           Tomatidine 1.00000000
21    Dehydrotomatidine TotalDehydrotomatine 0.03249178
22           Tomatidine TotalDehydrotomatine 0.37446722
23 TotalDehydrotomatine TotalDehydrotomatine 1.00000000</code></pre>
</div>
</div>
</section>
<section id="lower-triangle" class="level3">
<h3 class="anchored" data-anchor-id="lower-triangle">Lower triangle</h3>
<p>Create a lower triangle object to plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># "save as"</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>alkaloids_lower <span class="ot">&lt;-</span> alkaloids_cor</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="co"># use function upper.tri() and set the upper triangle all to NA</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># then we can keep only the lower triangle</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>alkaloids_lower[<span class="fu">upper.tri</span>(alkaloids_lower)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a><span class="co"># melt to go back to long format</span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>melted_alkaloids_lower <span class="ot">&lt;-</span> <span class="fu">melt</span>(alkaloids_lower, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a><span class="co"># did it work?</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(melted_alkaloids_lower) <span class="co"># yup</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  Var1              Var2       value
1    Dehydrotomatidine Dehydrotomatidine 1.000000000
2           Tomatidine Dehydrotomatidine 0.297446153
3 TotalDehydrotomatine Dehydrotomatidine 0.032491778
4             Tomatine Dehydrotomatidine 0.023823011
5 TotalHydroxytomatine Dehydrotomatidine 0.009912624
6 TotalAcetoxytomatine Dehydrotomatidine 0.032202892</code></pre>
</div>
</div>
<p>Plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remember we made alkaloid_labels</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(alkaloid_labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Dehydrotomatidine"                                 
 [2] "Tomatidine"                                        
 [3] "Dehydrotomatine"                                   
 [4] "Alpha-Tomatine"                                    
 [5] "Hydroxytomatine"                                   
 [6] "Acetoxytomatine"                                   
 [7] "Dehydrlycoperoside F, G, \nor Dehydroescueloside A"
 [8] "Lycoperoside F, G, \nor Escueloside A"             
 [9] "Escueloside B"                                     
[10] "Total Steroidal Alkaloids"                         </code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>melted_alkaloids_lower <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(value, <span class="dv">2</span>)), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"#f1a340"</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mid =</span> <span class="st">"#f7f7f7"</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">high =</span> <span class="st">"#998ec3"</span>,</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> alkaloid_labels) <span class="sc">+</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> alkaloid_labels) <span class="sc">+</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.7</span>),</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.direction =</span> <span class="st">"horizontal"</span>) <span class="sc">+</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Correlation </span><span class="sc">\n</span><span class="st">coefficient"</span>,</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span><span class="st">""</span>,</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Correlation between steroidal alkaloids using </span><span class="sc">\n</span><span class="st">Pearson's correlation coefficient"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: A numeric `legend.position` argument in `theme()` was deprecated in ggplot2
3.5.0.
â„¹ Please use the `legend.position.inside` argument of `theme()` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="08_correlations_files/figure-html/manual-lower-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="useful-resources" class="level2">
<h2 class="anchored" data-anchor-id="useful-resources">Useful resources</h2>
<ul>
<li><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a></li>
<li><a href="https://www.rdocumentation.org/packages/Hmisc/versions/4.7-1/topics/rcorr"><code>Hmisc::rcorr()</code></a></li>
<li><a href="https://www.rdocumentation.org/packages/ggcorrplot/versions/0.1.3/topics/ggcorrplot"><code>ggcorrplot::ggcorrplot()</code></a></li>
<li><a href="https://cran.r-project.org/web/packages/corrplot/vignettes/corrplot-intro.html"><code>corrplot::corrplot()</code></a></li>
<li><a href="https://ggobi.github.io/ggally/"><code>GGally</code></a></li>
<li><a href="https://ggobi.github.io/ggally/reference/ggcorr.html"><code>GGally:ggcorr()</code></a></li>
<li><a href="https://ggobi.github.io/ggally/reference/ggpairs.html"><code>GGally:ggpairs()</code></a></li>
</ul>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/www\.rdataviz\.com\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Created with RStudio and <a href="https://quarto.org/">Quarto</a> <br> Â© Jessica Cooperstone, 2025, License: <a href="https://creativecommons.org/licenses/by-nc/4.0/">CC BY-NC 4.0</a> <br> <a href="https://osu-codeclub.github.io/#category=website">Want to make a website like this one?</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/jcooperstone/rdataviz">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>